// @vtj/core/dist/index.mjs
import { uid as m, timestamp as G, isString as N, uuid as R, cloneDeep as T, upperFirstCamelCase as b, delay as O, merge as I, toArray as j, mitt as F } from "@vtj/base";
/**!
 * Copyright (c) 2026, VTJ.PRO All rights reserved.
 * @name @vtj/core 
 * @author CHC chenhuachun1549@dingtalk.com 
 * @version 0.16.21
 * @license <a href="https://vtj.pro/license.html">MIT License</a>
 */
const z = "0.16.21", W = "BuiltIn", H = "VueMaterial", L = "VueRouterMaterial", q = [H, L], Q = {
  vue: "Vue",
  "vue-router": "VueRouter"
}, X = {
  [H]: [
    "Transition",
    "TransitionGroup",
    "KeepAlive",
    "Teleport",
    "Suspense"
  ],
  [L]: ["RouterView", "RouterLink"]
}, Z = [
  "slot",
  "template",
  "component",
  "img",
  "div",
  "p",
  "h1",
  "h2",
  "h3",
  "span",
  "a"
], ee = [
  {
    label: "简体中文",
    value: "zh-CN"
  },
  {
    label: "English",
    value: "en"
  }
];
class te {
}
class ie {
  listeners = [];
  isReady = !1;
  triggerReady() {
    this.isReady = !0;
    for (const e of this.listeners)
      e();
    this.listeners = [];
  }
  ready(e) {
    this.isReady ? e() : this.listeners.push(e);
  }
  resetReady() {
    this.isReady = !1;
  }
}
class A {
  constructor(e, t, i) {
    this.name = e, this.value = t, this.defaultValue = i, this.setValue(t);
  }
  /**
   * 标识是否设置了值， 设置的值与默认值一致，表示未设置，在转换成dsl时会排查该属性
   */
  isUnset = !1;
  setValue(e) {
    this.value = e, this.isUnset = this.value === this.defaultValue;
  }
  getValue() {
    return this.value ?? this.defaultValue;
  }
  static toDsl(e = {}) {
    return Object.entries(e).reduce((t, [i, s]) => (s.isUnset || (t[i] = s.getValue()), t), {});
  }
  static parse(e = {}) {
    return Object.entries(e).reduce((t, [i, s]) => (t[i] = new A(i, s), t), {});
  }
}
class C {
  constructor(e) {
    this.schema = e;
    const { name: t, handler: i } = this.schema;
    this.name = t, this.handler = i, this.update(e);
  }
  name;
  handler;
  modifiers = {};
  update(e) {
    Object.assign(this.schema, e);
    const { handler: t, modifiers: i = {} } = this.schema;
    this.handler = t, this.modifiers = i;
  }
  static toDsl(e) {
    return Object.entries(e).reduce((t, [i, s]) => {
      const { handler: a, modifiers: r } = s;
      return t[i] = { name: i, handler: a, modifiers: r }, t;
    }, {});
  }
  static parse(e = {}) {
    return Object.entries(e).reduce(
      (t, [i, s]) => (t[i] = new C(s), t),
      {}
    );
  }
}
class v {
  constructor(e) {
    this.schema = e, this.id = e.id || m(), this.update(e);
  }
  /**
   * 标识
   */
  id;
  /**
   * 指令名称
   */
  name = "";
  /**
   * 参数
   */
  arg;
  /**
   * 修饰符
   */
  modifiers;
  /**
   * 指令值
   */
  value;
  /**
   *  v-for 迭代器
   */
  iterator;
  update(e) {
    Object.assign(this.schema, e);
    const { name: t, arg: i, modifiers: s, value: a, iterator: r } = this.schema;
    this.name = t, this.arg = i, this.modifiers = s, this.value = a, this.iterator = r;
  }
  static parse(e = []) {
    return e.map((t) => new v(t));
  }
  static toDsl(e = []) {
    return e.map((t) => {
      const { name: i, arg: s, modifiers: a, value: r, iterator: o, id: h } = t;
      return {
        id: h,
        name: i,
        arg: s,
        modifiers: a,
        value: r,
        iterator: o
      };
    });
  }
}
const c = "EVENT_NODE_CHANGE";
class _ {
  constructor(e, t = null) {
    this.parent = t;
    const { id: i = m(), name: s, from: a = "" } = e;
    this.id = i, this.name = s, this.from = a, this.update(e, !0), _.nodes[this.id] = this;
  }
  /**
   * 标记
   */
  __VTJ_NODE__ = !0;
  /**
   * 锁定
   */
  locked = !1;
  /**
   * 记录所有节点的实例
   */
  static nodes = {};
  /**
   * 节点唯一标识
   */
  id;
  /**
   * 名称，即组件的名称或html的标签名
   */
  name;
  /**
   * 组件来源
   */
  from;
  /**
   * 是否不可见
   */
  invisible = !1;
  /**
   * 子节点
   */
  children = "";
  /**
   * 放置在父组件的插槽
   */
  slot;
  /**
   * 节点属性
   */
  props = {};
  /**
   * 节点事件
   */
  events = {};
  /**
   * 指令
   */
  directives = [];
  /**
   * 销毁标识
   */
  disposed = !1;
  /**
   * 更新节点属性
   * @param schema
   * @param silent 是否静默，静默更新即不触发事件
   */
  update(e, t = !1) {
    const {
      invisible: i = !1,
      locked: s = !1,
      children: a = [],
      slot: r,
      props: o = {},
      events: h = {},
      directives: p = []
    } = e;
    this.invisible = i, this.locked = s, this.setChildren(a, !0), this.setSlot(r, !0), this.props = A.parse(o), this.events = C.parse(h), this.directives = v.parse(p), t || n.emit(c, this);
  }
  /**
   * 设置子节点
   * @param children
   * @param silent
   */
  setChildren(e = "", t = !1) {
    Array.isArray(e) ? this.children = e.map((i) => new _(i, this)) : this.children = e, t || n.emit(c, this);
  }
  /**
   * 设置节点放置的插槽
   * @param slot
   * @param silent
   */
  setSlot(e, t = !1) {
    this.slot = typeof e == "string" ? { name: e, params: [] } : e, t || n.emit(c, this);
  }
  /**
   * 新增或更新属性
   * @param name
   * @param value
   * @param defaultValue
   * @param silent
   */
  setProp(e, t, i, s = !1) {
    const a = this.props[e];
    a ? a.setValue(t) : this.props[e] = new A(e, t, i), s || n.emit(c, this);
  }
  /**
   * 删除属性
   * @param name
   * @param silent
   */
  removeProp(e, t = !1) {
    delete this.props[e], t || n.emit(c, this);
  }
  /**
   * 获取属性值
   * @param name
   * @returns
   */
  getPropValue(e) {
    const t = this.props[e];
    if (t)
      return t.getValue();
  }
  /**
   * 新增或更新事件
   * @param scheam
   * @param silent
   */
  setEvent(e, t = !1) {
    const i = this.events[e.name];
    i ? i.update(e) : this.events[e.name] = new C(e), t || n.emit(c, this);
  }
  /**
   * 删除事件
   * @param name
   * @param silent
   */
  removeEvent(e, t = !1) {
    delete this.events[e], t || n.emit(c, this);
  }
  /**
   * 新增或更新指令
   * @param scheam
   * @param silent
   */
  setDirective(e, t = !1) {
    const i = e instanceof v ? e : new v(e), s = this.directives.findIndex((a) => a.id === e.id);
    s >= 0 ? this.directives.splice(s, 1, i) : this.directives.push(i), t || n.emit(c, this);
  }
  /**
   * 删除指令
   * @param dirctive
   * @param silent
   */
  removeDirective(e, t = !1) {
    const i = this.directives.findIndex(
      (s) => s === e || s.id === e.id
    );
    i >= 0 && this.directives.splice(i, 1), t || n.emit(c, this);
  }
  /**
   * 删除子节点
   * @param node
   * @param silent
   * @returns
   */
  removeChild(e, t = !1) {
    const { children: i, disposed: s } = this;
    if (s || !Array.isArray(i)) return;
    const a = i.findIndex((r) => r === e);
    e.parent = null, i.splice(a, 1), t || n.emit(c, this);
  }
  /**
   * 追加子节点
   * @param node
   * @param silent
   * @returns
   */
  appendChild(e, t = !1) {
    const { children: i, disposed: s } = this;
    s || (e.parent = this, Array.isArray(i) ? i.push(e) : this.children = [e], t || n.emit(c, this));
  }
  /**
   * 在当前节点的后面插入节点
   * @param node
   * @param silent
   * @returns
   */
  insertAfter(e, t = !1) {
    if (!this.parent) return;
    const i = this.parent.children;
    if (Array.isArray(i)) {
      e.parent = this.parent;
      const s = i.indexOf(this);
      i.splice(s + 1, 0, e), t || n.emit(c, this.parent);
    }
  }
  /**
   * 在当前节点的前面插入节点
   * @param node
   * @param silent
   * @returns
   */
  insertBefore(e, t = !1) {
    if (!this.parent) return;
    const i = this.parent.children;
    if (Array.isArray(i)) {
      e.parent = this.parent;
      const s = i.indexOf(this);
      i.splice(s, 0, e), t || n.emit(c, this.parent);
    }
  }
  movePrev(e = !1) {
    const t = this.parent;
    if (!t) return;
    const i = t.children;
    if (Array.isArray(i)) {
      const s = i.indexOf(this);
      s > 0 && (i.splice(s, 1), i.splice(s - 1, 0, this), e || n.emit(c, t));
    }
  }
  moveNext(e = !1) {
    const t = this.parent;
    if (!t) return;
    const i = t.children;
    if (Array.isArray(i)) {
      const s = i.indexOf(this);
      s > -1 && s < i.length - 1 && (i.splice(s, 1), i.splice(s + 1, 0, this), e || n.emit(c, t));
    }
  }
  /**
   * 获取DSL
   * @returns
   */
  toDsl() {
    const {
      id: e,
      name: t,
      from: i,
      invisible: s,
      locked: a,
      slot: r,
      children: o,
      props: h,
      directives: p,
      events: P
    } = this, E = Array.isArray(o) ? o.map((U) => U.toDsl()) : o;
    return {
      id: e,
      name: t,
      from: i,
      invisible: s,
      locked: a,
      slot: r,
      children: E,
      props: A.toDsl(h),
      directives: v.toDsl(p),
      events: C.toDsl(P)
    };
  }
  /**
   * 销毁
   * @param silent
   * @returns
   */
  dispose(e = !1) {
    const { children: t, disposed: i } = this;
    i || (Array.isArray(t) && t.forEach((s) => s.dispose(!0)), this.parent ? this.parent.removeChild(this, e) : e || n.emit(c, this), this.parent = null, this.disposed = !0, delete _.nodes[this.id]);
  }
  lock(e = !1) {
    if (this.locked = !0, Array.isArray(this.children))
      for (const t of this.children)
        t.lock(!0);
    e || n.emit(c, this);
  }
  unlock(e = !1) {
    if (this.locked = !1, Array.isArray(this.children))
      for (const t of this.children)
        t.unlock(!0);
    e || n.emit(c, this);
  }
  setVisible(e, t = !1) {
    if (this.invisible = !e, Array.isArray(this.children))
      for (const i of this.children)
        i.setVisible(e, !0);
    t || n.emit(c, this);
  }
  isChild(e) {
    let t = !1;
    if (Array.isArray(this.children)) {
      for (const i of this.children)
        if (e === i || e.id === i.id) {
          t = !0;
          break;
        } else if (t = i.isChild(e), t)
          break;
    }
    return t;
  }
  findChildIndex(e) {
    return Array.isArray(this.children) ? this.children.findIndex((t) => t === e || t.id === e.id) : -1;
  }
}
const l = "EVENT_BLOCK_CHANGE";
class u {
  __VTJ_BLOCK__ = !0;
  id;
  name = "";
  inject = [];
  state = {};
  lifeCycles = {};
  methods = {};
  computed = {};
  watch = [];
  css = "";
  props = [];
  emits = [];
  expose = [];
  slots = [];
  dataSources = {};
  nodes = [];
  locked = !1;
  disposed = !1;
  static normalAttrs = [
    "name",
    "locked",
    "inject",
    "state",
    "lifeCycles",
    "methods",
    "computed",
    "watch",
    "css",
    "props",
    "emits",
    "expose",
    "slots",
    "dataSources",
    "__TEMPLATE_ID__"
  ];
  constructor(e) {
    const { id: t } = e;
    this.id = t || m(), this.update(e, !0);
  }
  update(e, t = !1) {
    for (const s of u.normalAttrs) {
      const a = e[s];
      a !== void 0 && (this[s] = a);
    }
    const { nodes: i = [] } = e;
    i.length ? this.nodes = i.map((s) => new _(s)) : this.nodes = [], t || n.emit(l, this);
  }
  /**
   * 获取DSL
   * @returns
   */
  toDsl(e) {
    const { __VTJ_BLOCK__: t, id: i, nodes: s } = this;
    return {
      ...u.normalAttrs.reduce(
        (r, o) => (r[o] = this[o], r),
        {}
      ),
      __VTJ_BLOCK__: t,
      __VERSION__: e || G().toString(),
      id: i,
      nodes: s.map((r) => r.toDsl())
    };
  }
  /**
   * 销毁
   */
  dispose() {
    this.nodes.map((e) => e.dispose(!0)), this.nodes = [], this.disposed = !0;
  }
  /**
   * 设置通用函数属性
   * @param type
   * @param name
   * @param value
   * @param silent
   */
  setFunction(e, t, i, s = !1) {
    this[e][t] = i, s || n.emit(l, this);
  }
  /**
   * 删除通用函数属性
   * @param type
   * @param name
   * @param silent
   */
  removeFunction(e, t, i = !1) {
    delete this[e][t], i || n.emit(l, this);
  }
  /**
   * 设置状态
   * @param name
   * @param value
   * @param silent
   */
  setState(e, t, i = !1) {
    this.state[e] = t, i || n.emit(l, this);
  }
  /**
   * 删除状态
   * @param name
   * @param silent
   */
  removeState(e, t = !1) {
    delete this.state[e], t || n.emit(l, this);
  }
  /**
   * 更新CSS
   * @param content
   * @param silent
   */
  setCss(e, t = !1) {
    this.css = e, t || n.emit(l, this);
  }
  /**
   * 新增或更新 watch
   * @param watch
   * @param silent
   */
  setWatch(e, t = !1) {
    e.id = e.id || m();
    const i = this.watch.findIndex(
      (s) => s.id && s.id === e.id || s === e
    );
    i > -1 ? this.watch.splice(i, 1, e) : this.watch.push(e), t || n.emit(l, this);
  }
  /**
   * 删除 watch
   * @param watch
   * @param silent
   */
  removeWatch(e, t = !1) {
    const i = this.watch.findIndex(
      (s) => s.id && s.id === e.id || s === e
    );
    i > -1 && (this.watch.splice(i, 1), t || n.emit(l, this));
  }
  /**
   * 定义属性参数
   * @param prop
   * @param silent
   */
  setProp(e, t = !1) {
    const i = this.props.findIndex(
      (s) => typeof s == "string" ? s === e.name : s.name === e.name
    );
    i > -1 ? this.props.splice(i, 1, e) : this.props.push(e), t || n.emit(l, this);
  }
  /**
   * 删除属性
   * @param prop
   * @param silent
   */
  removeProp(e, t = !1) {
    const i = this.props.findIndex(
      (s) => typeof s == "string" ? s === e.name : s.name === e.name
    );
    i > -1 && (this.props.splice(i, 1), t || n.emit(l, this));
  }
  /**
   * 设置事件
   * @param emit
   * @param silent
   */
  setEmit(e, t = !1) {
    const i = N(e) ? { name: e, params: [] } : e, s = this.emits.findIndex(
      (a) => a === i.name || a.name === i.name
    );
    s > -1 ? this.emits.splice(s, 1, i) : this.emits.push(i), t || n.emit(l, this);
  }
  /**
   * 删除事件
   * @param emit
   * @param silent
   */
  removeEmit(e, t = !1) {
    const i = this.emits.findIndex(
      (s) => N(s) ? s === e : s.name === e
    );
    i > -1 && (this.emits.splice(i, 1), t || n.emit(l, this));
  }
  setExpose(e, t = !1) {
    this.expose = e, t || n.emit(l, this);
  }
  /**
   * 设置插槽
   * @param slot
   * @param silent
   */
  setSlot(e, t = !1) {
    const i = N(e) ? { name: e, params: [] } : e, s = this.slots.findIndex(
      (a) => a === i.name || a.name === i.name
    );
    s > -1 ? this.slots.splice(s, 1, i) : this.slots.push(i), t || n.emit(l, this);
  }
  /**
   * 删除插槽
   * @param slot
   * @param silent
   */
  removeSlot(e, t = !1) {
    const i = this.slots.findIndex(
      (s) => N(s) ? s === e : s.name === e
    );
    i > -1 && (this.slots.splice(i, 1), t || n.emit(l, this));
  }
  /**
   * 设置注入
   * @param inject
   * @param silent
   */
  setInject(e, t = !1) {
    const i = this.inject.findIndex((s) => s.name === e.name);
    i > -1 ? this.inject.splice(i, 1, e) : this.inject.push(e), t || n.emit(l, this);
  }
  /**
   * 删除注入
   * @param inject
   * @param silent
   */
  removeInject(e, t = !1) {
    const i = this.inject.findIndex((s) => s.name === e.name);
    i > -1 && (this.inject.splice(i, 1), t || n.emit(l, this));
  }
  /**
   * 设置数据源
   * @param source
   * @param silent
   */
  setDataSource(e, t = !1) {
    this.dataSources[e.name] = e, t || n.emit(l, this);
  }
  /**
   * 删除数据源
   * @param name
   * @param silent
   */
  removeDataSource(e, t = !1) {
    delete this.dataSources[e], t || n.emit(l, this);
  }
  insertAfter(e, t, i = !1) {
    e.parent = null;
    const s = this.nodes.indexOf(t);
    this.nodes.splice(s + 1, 0, e), i || n.emit(l, this);
  }
  insertBefore(e, t, i = !1) {
    e.parent = null;
    const s = this.nodes.indexOf(t);
    this.nodes.splice(s, 0, e), i || n.emit(l, this);
  }
  appendNode(e, t = !1) {
    e.parent = null, this.nodes.push(e), t || n.emit(l, this);
  }
  /**
   * 添加节点
   * @param node
   * @param target
   * @param position
   * @param silent
   */
  addNode(e, t, i = "inner", s = !1) {
    t ? ["left", "top"].includes(i) ? t.parent ? t.insertAfter(e, s) : this.insertBefore(e, t, s) : ["right", "bottom"].includes(i) ? t.parent ? t.insertAfter(e, s) : this.insertAfter(e, t, s) : t.appendChild(e, s) : this.appendNode(e, s);
  }
  __removeNode(e, t = !1) {
    const i = this.nodes.findIndex((s) => s.id === e.id);
    i > -1 && (this.nodes.splice(i, 1), t || n.emit(l, this));
  }
  /**
   * 删除节点
   * @param node
   * @param silent
   */
  removeNode(e, t = !1) {
    e.parent ? e.dispose(t) : (e.dispose(!0), this.__removeNode(e, t));
  }
  /**
   * 移动节点
   * @param node
   * @param target
   * @param position
   * @param silent
   */
  move(e, t, i = "inner", s = !1) {
    e.parent ? e.parent.removeChild(e, !0) : this.__removeNode(e, !0), this.addNode(e, t, i, s);
  }
  /**
   * 向前交换节点
   * @param node
   * @param silent
   */
  movePrev(e, t = !1) {
    if (e.parent)
      e.movePrev(t);
    else {
      const i = this.nodes, s = i.indexOf(e);
      s > 0 && (i.splice(s, 1), i.splice(s - 1, 0, e), t || n.emit(l, this));
    }
  }
  /**
   * 向后交换节点
   * @param node
   * @param silent
   */
  moveNext(e, t = !1) {
    if (e.parent)
      e.moveNext(t);
    else {
      const i = this.nodes, s = i.indexOf(e);
      s > -1 && s < i.length - 1 && (i.splice(s, 1), i.splice(s + 1, 0, e), t || n.emit(l, this));
    }
  }
  /**
   * 克隆节点
   * @param target
   * @param silent
   * @returns
   */
  cloneNode(e, t = !1) {
    const i = J(e.toDsl()), s = new _(i);
    return this.addNode(s, e, "bottom", t), s;
  }
  lock(e = !1) {
    this.locked = !0;
    for (const t of this.nodes)
      t.lock(!0);
    e || n.emit(l, this);
  }
  unlock(e = !1) {
    this.locked = !1;
    for (const t of this.nodes)
      t.unlock(!0);
    e || n.emit(l, this);
  }
  isChild(e) {
    let t = !1;
    for (const i of this.nodes)
      if (e === i || e.id === i.id) {
        t = !0;
        break;
      } else if (t = i.isChild(e), t)
        break;
    return t;
  }
}
const d = "EVENT_PROJECT_CHANGE", D = "EVENT_PROJECT_ACTIVED", B = "EVENT_PROJECT_DEPS_CHANGE", g = "EVENT_PROJECT_PAGES_CHANGE", y = "EVENT_PROJECT_BLOCKS_CHANGE", V = "EVENT_PROJECT_APIS_CHANGE", w = "EVENT_PROJECT_META_CHANGE", $ = "EVENT_PROJECT_PUBLISH", M = "EVENT_PROJECT_FILE_PUBLISH", K = "EVENT_PROJECT_GEN_SOURCE";
class k {
  id = "";
  locked = "";
  platform = "web";
  name = "";
  description = "";
  homepage = "";
  dependencies = [];
  pages = [];
  blocks = [];
  apis = [];
  meta = [];
  currentFile = null;
  config = {};
  uniConfig = {};
  globals = {};
  i18n = {
    locale: "zh-CN",
    fallbackLocale: "zh-CN",
    messages: []
  };
  env = [];
  __BASE_PATH__ = "/";
  __UID__ = R(!0);
  static attrs = [
    "locked",
    "platform",
    "name",
    "homepage",
    "description",
    "dependencies",
    "pages",
    "blocks",
    "apis",
    "meta",
    "config",
    "uniConfig",
    "globals",
    "i18n",
    "env",
    "__BASE_PATH__",
    "__UID__"
  ];
  constructor(e) {
    const { id: t, __UID__: i } = e;
    this.id = t || m(), this.__UID__ = i || R(!0), this.update(e, !0);
  }
  update(e, t = !1) {
    for (const i of k.attrs) {
      const s = e[i];
      s && (this[i] = s);
    }
    t || n.emit(d, {
      model: this,
      type: "update",
      data: e
    });
  }
  isPageFile(e) {
    return e.type === "page";
  }
  cleanPagesDsl(e) {
    for (const t of e)
      delete t.dsl, t.children && t.children.length && this.cleanPagesDsl(t.children);
  }
  toDsl(e) {
    const { id: t } = this, i = k.attrs.reduce(
      (s, a) => (s[a] = this[a], s),
      {}
    );
    return i.pages && (i.pages = i.pages.map((s) => T({
      ...s,
      dsl: void 0
    })), this.cleanPagesDsl(i.pages)), i.blocks && (i.blocks = i.blocks.map((s) => T({
      ...s,
      dsl: void 0
    }))), {
      __VTJ_PROJECT__: !0,
      id: t,
      ...i
    };
  }
  /**
   * 打开文件
   * @param file
   * @param silent
   */
  active(e, t = !1) {
    this.currentFile = e, t || n.emit(D, {
      model: this,
      type: "update",
      data: e
    });
  }
  /**
   * 关闭文件
   * @param silent
   */
  deactivate(e = !1) {
    this.currentFile = null, e || n.emit(D, {
      model: this,
      type: "update",
      data: null
    });
  }
  /**
   * 新增或更新依赖
   * @param item
   * @param silent
   */
  setDeps(e, t = !1) {
    const i = this.dependencies, s = i.findIndex((r) => r.package === e.package);
    let a;
    if (s > -1 ? (a = "update", i.splice(s, 1, {
      ...i[s],
      ...e
    })) : (a = "create", i.push(e)), !t) {
      const r = {
        model: this,
        type: a,
        data: e
      };
      n.emit(B, r), n.emit(d, r);
    }
  }
  /**
   * 删除依赖
   * @param item
   * @param silent
   */
  removeDeps(e, t = !1) {
    const i = this.dependencies, s = i.findIndex((a) => a.package === e.package);
    if (s > -1 && i.splice(s, 1), !t) {
      const a = {
        model: this,
        type: "delete",
        data: e
      };
      n.emit(B, a), n.emit(d, a);
    }
  }
  /**
   * 根据页面id查找页面或目录
   * @param id
   * @returns
   */
  getPage(e) {
    const t = (i, s = []) => {
      for (const a of s) {
        if (a.id === i)
          return a;
        if (a.children && a.children.length) {
          const r = t(i, a.children);
          if (r)
            return r;
        }
      }
    };
    return t(e, this.pages);
  }
  /**
   * 查找全部页面，不含目录
   * @returns
   */
  getPages() {
    const e = (t = []) => {
      let i = [];
      for (const s of t)
        s.dir || s.layout ? (s.layout && i.push(s), s.children && s.children.length && (i = i.concat(e(s.children)))) : i.push(s);
      return i;
    };
    return e(this.pages);
  }
  getHomepage() {
    return this.homepage ? this.getPage(this.homepage) : this.getPages()[0];
  }
  /**
   * 新建页面
   * @param page
   * @param parentId
   * @param silent
   */
  async createPage(e, t, i = !1) {
    if (e.id = e.raw ? e.name : e.id || m(), e.type = "page", e.dir ? e.children = [] : e.dsl = e.dsl || new u({
      id: e.id,
      name: b(e.name)
    }).toDsl(), t) {
      const s = this.getPage(t);
      s ? s.children ? s.children.push(e) : s.children = [e] : console.warn(`not found PageFile for id: ${t} `);
    } else
      this.pages.push(e);
    if (!i) {
      const s = {
        model: this,
        type: "create",
        data: e
      };
      n.emit(g, s), n.emit(d, s);
    }
    return !this.currentFile && !e.dir && (await O(1e3), this.active(e, i)), e;
  }
  /**
   * 更新页面
   * @param page
   * @param silent
   */
  updatePage(e, t = !1) {
    const i = this.getPage(e.id);
    if (delete e.dsl, i ? Object.assign(i, e) : console.warn(`not found PageFile for id: ${e.id} `), !t) {
      const s = {
        model: this,
        type: "update",
        data: e
      };
      n.emit(g, s), n.emit(d, s);
    }
    return i;
  }
  findParent(e) {
    const t = (i, s = [], a) => {
      for (const r of s) {
        if (r.id === i)
          return a;
        if (r.children && r.children.length) {
          const o = t(i, r.children, r);
          if (o)
            return o;
        }
      }
    };
    return t(e, this.pages);
  }
  movePageTo(e, t, i = !1) {
    const s = t ? this.getPage(t) : null, a = this.getPage(e);
    if (!a) return !1;
    const r = this.findParent(e);
    if (r ? r.children = r.children?.filter(
      (o) => o.id !== e
    ) : this.pages = this.pages.filter((o) => o.id !== e), s ? (s.dir || s.layout) && (s.children ? s.children.push(a) : s.children = [a]) : this.pages.push(a), !i) {
      const o = {
        model: this,
        type: "update",
        data: a
      };
      n.emit(g, o), n.emit(d, o);
    }
    return !0;
  }
  getPageTree() {
    const e = T(this.pages);
    return this.cleanPagesDsl(e), e;
  }
  /**
   * 复制页面
   * @param page
   * @param parentId
   * @param silent
   */
  clonePage(e, t, i = !1) {
    const s = m(), a = `${e.name}Copy`, r = `${e.title}_副本`, o = new u({
      id: s,
      name: a
    }).toDsl(), h = I({}, e, { id: s, name: a, title: r, dsl: o }), p = t ? this.getPage(t)?.children || [] : this.pages, P = p.findIndex((E) => E.id === e.id);
    if (p.splice(P + 1, 0, h), !i) {
      const E = {
        model: this,
        type: "clone",
        data: {
          source: e,
          target: h
        }
      };
      n.emit(g, E), n.emit(d, E);
    }
  }
  async saveToBlock(e, t = !1) {
    this.active(e, t), await O(1e3);
    const i = m(), s = e.name, a = e.title, r = new u({
      ...e.dsl,
      id: i,
      name: s
    }).toDsl(), o = I({}, e, {
      id: i,
      name: s,
      title: a,
      dsl: r,
      type: "block",
      fromType: "Schema"
    });
    if (this.blocks.push(o), !t) {
      const h = {
        model: this,
        type: "create",
        data: o
      };
      n.emit(y, h), n.emit(d, h);
    }
  }
  /**
   * 删除页面或目录
   * @param id
   * @param silent
   */
  removePage(e, t = !1) {
    const i = this.getPage(e), s = (a, r) => {
      const o = r.findIndex((h) => h.id === a);
      if (o >= 0) {
        r.splice(o, 1);
        return;
      }
      for (const h of r)
        h.children && h.children.length && s(a, h.children);
    };
    if (s(e, this.pages), e === this.homepage && (this.homepage = ""), this.currentFile?.id === e && this.deactivate(t), !t) {
      const a = {
        model: this,
        type: "delete",
        data: i
      };
      n.emit(g, a), n.emit(d, a);
    }
  }
  /**
   * 获取区块文件
   * @param id
   * @returns
   */
  getBlock(e) {
    return this.blocks.find((t) => t.id === e);
  }
  getFile(e) {
    return this.getPage(e) || this.getBlock(e);
  }
  /**
   * 创建区块
   * @param block
   * @param silent
   */
  async createBlock(e, t = !1) {
    const i = e.id || m(), s = b(e.name);
    e.id = i, e.type = "block", e.fromType = e.fromType || "Schema", e.dsl = new u({ id: i, name: s }).toDsl(), this.blocks.push(e);
    const a = e.fromType || "Schema";
    if (!t) {
      const r = {
        model: this,
        type: "create",
        data: e
      };
      n.emit(y, r), n.emit(d, r);
    }
    return !this.currentFile && a === "Schema" && (await O(1e3), this.active(e, t)), e;
  }
  /**
   *
   * @param block 更新区块
   * @param silent
   */
  updateBlock(e, t = !1) {
    const i = this.getBlock(e.id);
    if (i ? (Object.assign(i, e), i.dsl && (i.dsl.name = e.name)) : console.warn(`not found PageFile for id: ${e.id} `), !t) {
      const s = {
        model: this,
        type: "update",
        data: e
      };
      n.emit(y, s), n.emit(d, s);
    }
    return i;
  }
  cloneBlock(e, t = !1) {
    const i = m(), s = `${e.name}Copy`, a = `${e.title}_副本`, r = new u({
      id: i,
      name: s
    }).toDsl(), o = I({}, e, { id: i, name: s, title: a, dsl: r }), h = this.blocks.findIndex((p) => p.id === e.id);
    if (this.blocks.splice(h + 1, 0, o), !t) {
      const p = {
        model: this,
        type: "clone",
        data: {
          source: e,
          target: o
        }
      };
      n.emit(y, p), n.emit(d, p);
    }
  }
  /**
   * 删除区块
   * @param id
   * @param silent
   */
  removeBlock(e, t = !1) {
    const i = this.getBlock(e), s = this.blocks, a = s.findIndex((r) => r.id === e);
    if (a > -1 ? (s.splice(a, 1), this.currentFile?.id === e && this.deactivate(t)) : console.warn(`not found PageFile for id: ${e} `), !t) {
      const r = {
        model: this,
        type: "delete",
        data: i
      };
      n.emit(y, r), n.emit(d, r);
    }
  }
  /**
   * 检查是否存在名称的区块
   * @param name
   * @param excludes
   * @returns
   */
  existBlockName(e, t = []) {
    return this.blocks.some((i) => i.name === e && !t.includes(i.id));
  }
  /**
   * 检测是否存在名称的页面
   * @param name
   * @param excludes
   * @returns
   */
  existPageName(e, t = []) {
    return this.getPages().some((s) => s.name === e && !t.includes(s.id));
  }
  /**
   * 新增或更新api
   * @param item
   * @param silent
   */
  setApi(e, t = !1) {
    const i = this.apis.find(
      (a) => a.name === e.name || a.id === e.id
    );
    let s;
    if (i ? (s = "update", Object.assign(i, e)) : (s = "create", e.id = m(), this.apis.push(e)), !t) {
      const a = {
        model: this,
        type: s,
        data: e
      };
      n.emit(V, a), n.emit(d, a);
    }
    return i || e;
  }
  setApis(e, t = !1) {
    for (const i of e)
      this.setApi(i, !0);
    if (!t) {
      const i = {
        model: this,
        type: "create",
        data: e
      };
      n.emit(V, i), n.emit(d, i);
    }
  }
  /**
   * 删除api
   * @param name
   * @param silent
   */
  removeApi(e, t = !1) {
    const i = this.apis.findIndex((s) => s.name === e || s.id === e);
    if (i > -1 ? this.apis.splice(i, 1) : console.warn(`not found Api for name: ${e} `), !t) {
      const s = {
        model: this,
        type: "delete",
        data: e
      };
      n.emit(V, s), n.emit(d, s);
    }
  }
  existApiName(e, t = []) {
    return this.apis.some((i) => i.name === e && !t.includes(i.id));
  }
  setMeta(e, t = !1) {
    const i = this.meta.find(
      (a) => a.code === e.code || a.id === e.id
    );
    let s;
    if (i ? (s = "update", Object.assign(i, e)) : (s = "create", e.id = m(), this.meta.push(e)), !t) {
      const a = {
        model: this,
        type: s,
        data: e
      };
      n.emit(w, a), n.emit(d, a);
    }
  }
  removeMeta(e, t = !1) {
    const i = this.meta.findIndex((s) => s.code === e || s.id === e);
    if (i > -1 ? this.meta.splice(i, 1) : console.warn(`not found meta for name: ${name} `), !t) {
      const s = {
        model: this,
        type: "delete",
        data: e
      };
      n.emit(w, s), n.emit(d, s);
    }
  }
  existMetaCode(e, t = []) {
    return this.meta.some((i) => i.code === e && !t.includes(i.id));
  }
  setHomepage(e, t = !1) {
    if (this.homepage = e, !t) {
      const i = {
        model: this,
        type: "update",
        data: e
      };
      n.emit(d, i);
    }
  }
  setConfig(e, t = !1) {
    if (this.config = Object.assign(this.config, e), !t) {
      const i = {
        model: this,
        type: "update",
        data: e
      };
      n.emit(d, i);
    }
  }
  setUniConfig(e, t, i = !1) {
    if (this.uniConfig = Object.assign(this.uniConfig, { [e]: t }), !i) {
      const s = {
        model: this,
        type: "update",
        data: this.uniConfig
      };
      n.emit(d, s);
    }
  }
  setGloblas(e, t, i = !1) {
    if (this.globals = Object.assign(this.globals, { [e]: t }), !i) {
      const s = {
        model: this,
        type: "update",
        data: this.globals
      };
      n.emit(d, s);
    }
  }
  setI18n(e, t = !1) {
    if (this.i18n = Object.assign(this.i18n, e), !t) {
      const i = {
        model: this,
        type: "update",
        data: this.i18n
      };
      n.emit(d, i);
    }
  }
  setEnv(e, t = !1) {
    if (this.env = [...e], !t) {
      const i = {
        model: this,
        type: "update",
        data: this.env
      };
      n.emit(d, i);
    }
  }
  publish(e) {
    const t = {
      model: this,
      type: "publish",
      data: e || this
    };
    e ? n.emit(M, t) : n.emit($, t);
  }
  genSource() {
    const e = {
      model: this,
      type: "gen",
      data: null
    };
    n.emit(K, e);
  }
  lock(e) {
    this.locked = e;
    const t = {
      model: this,
      type: "update",
      data: e
    };
    n.emit(d, t);
  }
  unlock(e) {
    if (e !== this.locked)
      return;
    this.locked = "";
    const t = {
      model: this,
      type: "update",
      data: null
    };
    n.emit(d, t);
  }
  getPageRoutes(e, t) {
    const i = this.platform === "uniapp", s = e || (i ? "pages" : "page"), a = this.getPages(), r = t || "";
    return a.map((o) => ({
      id: o.id,
      path: `${r}/${s}/${o.id}`,
      name: o.name,
      title: o.title,
      meta: o.meta
    }));
  }
}
const x = "EVENT_HISTORY_CHANGE", S = "EVENT_HISTORY_LOAD";
class se {
  options = { max: 50 };
  index = -1;
  id;
  items;
  constructor(e, t = {}) {
    Object.assign(this.options, t);
    const { id: i, items: s = [] } = e;
    this.id = i, this.items = s;
  }
  toDsl() {
    const { id: e, items: t } = this;
    return {
      id: e,
      items: t.map((i) => ({
        id: i.id,
        label: i.label,
        remark: i.remark || ""
      }))
    };
  }
  /**
   * 获取历史项
   * @param id
   * @returns
   */
  get(e) {
    return this.items.find((t) => t.id === e);
  }
  /**
   * 增加历史记录
   * @param dsl
   * @param silent
   */
  add(e, t = "", i = !1) {
    const { max: s } = this.options, a = {
      id: m(),
      label: (/* @__PURE__ */ new Date()).toLocaleString(),
      remark: t,
      dsl: T(e)
    };
    this.items.unshift(a);
    const r = this.items.filter((o) => !o.remark);
    if (r.length > s) {
      const o = r.splice(s).map((h) => h.id);
      this.items = this.items.filter((h) => !o.includes(h.id)), i || n.emit(x, {
        model: this,
        type: "delete",
        data: o
      });
    }
    this.index = -1, i || n.emit(x, {
      model: this,
      type: "create",
      data: a
    });
  }
  /**
   * 更新历史记录
   * @param id
   * @param remark
   * @param silent
   */
  update(e, t = !1) {
    const i = this.items.find((s) => s.id === e.id);
    i && Object.assign(i, e), t || n.emit(x, {
      model: this,
      type: "update",
      data: i
    });
  }
  /**
   * 删除历史记录
   * @param id
   * @param silent
   */
  remove(e, t = !1) {
    const i = j(e);
    for (const s of i) {
      const a = this.items.findIndex((r) => r.id === s);
      a > -1 ? (this.items.splice(a, 1), a === this.index ? this.index = -1 : this.index >= this.items.length && (this.index = this.items.length - 1)) : console.warn(`not found HistoryItem for id: ${e} `);
    }
    t || n.emit(x, {
      model: this,
      type: "delete",
      data: i
    });
  }
  forward(e = !1) {
    const { index: t, items: i } = this;
    if (t < 0) return;
    --this.index;
    const s = i[this.index];
    s && !e && n.emit(S, {
      model: this,
      type: "load",
      data: s
    });
  }
  backward(e = !1) {
    const { index: t, items: i } = this;
    if (t >= i.length - 1) return;
    t < 0 && (this.index = 0), ++this.index;
    const s = i[this.index];
    s && !e && n.emit(S, {
      model: this,
      type: "load",
      data: s
    });
  }
  load(e, t = !1) {
    const i = this.items.findIndex((s) => s.id === e);
    i >= 0 && (this.index = i, t || n.emit(S, {
      model: this,
      type: "load",
      data: this.items[i]
    }));
  }
  clear(e = !1) {
    this.index = -1;
    const t = this.items.map((i) => i.id);
    this.items = [], e || n.emit(x, {
      model: this,
      type: "clear",
      data: t
    });
  }
}
const n = F();
function ne(f) {
  return f instanceof u;
}
function ae(f) {
  return f instanceof _;
}
function re(f) {
  return !!f.__VTJ_BLOCK__;
}
function J(f) {
  const e = T(f);
  return delete e.id, Array.isArray(e.children) && (e.children = e.children.map((t) => J(t))), e;
}
function oe(f) {
  const e = f.fromType || "Schema";
  return e === "Schema" ? {
    type: "Schema",
    id: f.id
  } : e === "UrlSchema" ? {
    type: "UrlSchema",
    url: (f.urls || "").split(",")[0] || ""
  } : e === "Plugin" ? {
    type: "Plugin",
    urls: (f.urls || "").split(","),
    library: f.library
  } : "";
}
export {
  X as BUILT_IN_COMPONENTS,
  Q as BUILT_IN_LIBRARAY_MAP,
  q as BUILT_IN_MATERIALS,
  W as BUILT_IN_NAME,
  Z as BUILT_IN_TAGS,
  H as BUILT_IN_VUE,
  L as BUILT_IN_VUE_ROUTER,
  ie as Base,
  u as BlockModel,
  v as DirectiveModel,
  l as EVENT_BLOCK_CHANGE,
  x as EVENT_HISTORY_CHANGE,
  S as EVENT_HISTORY_LOAD,
  c as EVENT_NODE_CHANGE,
  D as EVENT_PROJECT_ACTIVED,
  V as EVENT_PROJECT_APIS_CHANGE,
  y as EVENT_PROJECT_BLOCKS_CHANGE,
  d as EVENT_PROJECT_CHANGE,
  B as EVENT_PROJECT_DEPS_CHANGE,
  M as EVENT_PROJECT_FILE_PUBLISH,
  K as EVENT_PROJECT_GEN_SOURCE,
  w as EVENT_PROJECT_META_CHANGE,
  g as EVENT_PROJECT_PAGES_CHANGE,
  $ as EVENT_PROJECT_PUBLISH,
  C as EventModel,
  se as HistoryModel,
  ee as I18N_LOCALES,
  _ as NodeModel,
  k as ProjectModel,
  A as PropModel,
  te as Service,
  z as VTJ_CORE_VERSION,
  J as cloneDsl,
  oe as createNodeFrom,
  n as emitter,
  ne as isBlock,
  re as isBlockSchema,
  ae as isNode
};
