// @vtj/utils/dist/index.mjs
var kt = Object.defineProperty;
var Dt = (t, e, n) => e in t ? kt(t, e, { enumerable: !0, configurable: !0, writable: !0, value: n }) : t[e] = n;
var _ = (t, e, n) => Dt(t, typeof e != "symbol" ? e + "" : e, n);
import { omit as me, merge as W, debounce as Bt, throttle as vt, isUrl as It, pathToRegexpCompile as qt, uuid as Mt, rURL as Ht, template as $t } from "@vtj/base";
export * from "@vtj/base";
import { ref as ge } from "vue";
(function() {
  if (typeof window > "u" || typeof EventTarget > "u")
    return;
  const t = EventTarget.prototype.addEventListener;
  EventTarget.prototype.addEventListener = function(e, n, r) {
    typeof r != "boolean" && (r = r || {}, r.passive = !1), t.call(this, e, n, r);
  };
})();
/**!
 * Copyright (c) 2026, VTJ.PRO All rights reserved.
 * @name @vtj/utils 
 * @author CHC chenhuachun1549@dingtalk.com 
 * @version 0.16.21
 * @license <a href="https://vtj.pro/license.html">MIT License</a>
 */
const Zr = "0.16.21";
function at(t, e) {
  return function() {
    return t.apply(e, arguments);
  };
}
const { toString: Jt } = Object.prototype, { getPrototypeOf: _e } = Object, { iterator: ce, toStringTag: ct } = Symbol, le = /* @__PURE__ */ ((t) => (e) => {
  const n = Jt.call(e);
  return t[n] || (t[n] = n.slice(8, -1).toLowerCase());
})(/* @__PURE__ */ Object.create(null)), U = (t) => (t = t.toLowerCase(), (e) => le(e) === t), ue = (t) => (e) => typeof e === t, { isArray: H } = Array, M = ue("undefined");
function K(t) {
  return t !== null && !M(t) && t.constructor !== null && !M(t.constructor) && x(t.constructor.isBuffer) && t.constructor.isBuffer(t);
}
const lt = U("ArrayBuffer");
function Wt(t) {
  let e;
  return typeof ArrayBuffer < "u" && ArrayBuffer.isView ? e = ArrayBuffer.isView(t) : e = t && t.buffer && lt(t.buffer), e;
}
const zt = ue("string"), x = ue("function"), ut = ue("number"), X = (t) => t !== null && typeof t == "object", Vt = (t) => t === !0 || t === !1, se = (t) => {
  if (le(t) !== "object")
    return !1;
  const e = _e(t);
  return (e === null || e === Object.prototype || Object.getPrototypeOf(e) === null) && !(ct in t) && !(ce in t);
}, Kt = (t) => {
  if (!X(t) || K(t))
    return !1;
  try {
    return Object.keys(t).length === 0 && Object.getPrototypeOf(t) === Object.prototype;
  } catch {
    return !1;
  }
}, Xt = U("Date"), Gt = U("File"), Qt = (t) => !!(t && typeof t.uri < "u"), Yt = (t) => t && typeof t.getParts < "u", Zt = U("Blob"), en = U("FileList"), tn = (t) => X(t) && x(t.pipe);
function nn() {
  return typeof globalThis < "u" ? globalThis : typeof self < "u" ? self : typeof window < "u" ? window : typeof global < "u" ? global : {};
}
const qe = nn(), Me = typeof qe.FormData < "u" ? qe.FormData : void 0, rn = (t) => {
  let e;
  return t && (Me && t instanceof Me || x(t.append) && ((e = le(t)) === "formdata" || // detect form-data instance
  e === "object" && x(t.toString) && t.toString() === "[object FormData]"));
}, sn = U("URLSearchParams"), [on, an, cn, ln] = [
  "ReadableStream",
  "Request",
  "Response",
  "Headers"
].map(U), un = (t) => t.trim ? t.trim() : t.replace(/^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g, "");
function G(t, e, { allOwnKeys: n = !1 } = {}) {
  if (t === null || typeof t > "u")
    return;
  let r, s;
  if (typeof t != "object" && (t = [t]), H(t))
    for (r = 0, s = t.length; r < s; r++)
      e.call(null, t[r], r, t);
  else {
    if (K(t))
      return;
    const o = n ? Object.getOwnPropertyNames(t) : Object.keys(t), i = o.length;
    let a;
    for (r = 0; r < i; r++)
      a = o[r], e.call(null, t[a], a, t);
  }
}
function ft(t, e) {
  if (K(t))
    return null;
  e = e.toLowerCase();
  const n = Object.keys(t);
  let r = n.length, s;
  for (; r-- > 0; )
    if (s = n[r], e === s.toLowerCase())
      return s;
  return null;
}
const B = typeof globalThis < "u" ? globalThis : typeof self < "u" ? self : typeof window < "u" ? window : global, dt = (t) => !M(t) && t !== B;
function Se() {
  const { caseless: t, skipUndefined: e } = dt(this) && this || {}, n = {}, r = (s, o) => {
    if (o === "__proto__" || o === "constructor" || o === "prototype")
      return;
    const i = t && ft(n, o) || o;
    se(n[i]) && se(s) ? n[i] = Se(n[i], s) : se(s) ? n[i] = Se({}, s) : H(s) ? n[i] = s.slice() : (!e || !M(s)) && (n[i] = s);
  };
  for (let s = 0, o = arguments.length; s < o; s++)
    arguments[s] && G(arguments[s], r);
  return n;
}
const fn = (t, e, n, { allOwnKeys: r } = {}) => (G(
  e,
  (s, o) => {
    n && x(s) ? Object.defineProperty(t, o, {
      value: at(s, n),
      writable: !0,
      enumerable: !0,
      configurable: !0
    }) : Object.defineProperty(t, o, {
      value: s,
      writable: !0,
      enumerable: !0,
      configurable: !0
    });
  },
  { allOwnKeys: r }
), t), dn = (t) => (t.charCodeAt(0) === 65279 && (t = t.slice(1)), t), pn = (t, e, n, r) => {
  t.prototype = Object.create(e.prototype, r), Object.defineProperty(t.prototype, "constructor", {
    value: t,
    writable: !0,
    enumerable: !1,
    configurable: !0
  }), Object.defineProperty(t, "super", {
    value: e.prototype
  }), n && Object.assign(t.prototype, n);
}, hn = (t, e, n, r) => {
  let s, o, i;
  const a = {};
  if (e = e || {}, t == null) return e;
  do {
    for (s = Object.getOwnPropertyNames(t), o = s.length; o-- > 0; )
      i = s[o], (!r || r(i, t, e)) && !a[i] && (e[i] = t[i], a[i] = !0);
    t = n !== !1 && _e(t);
  } while (t && (!n || n(t, e)) && t !== Object.prototype);
  return e;
}, mn = (t, e, n) => {
  t = String(t), (n === void 0 || n > t.length) && (n = t.length), n -= e.length;
  const r = t.indexOf(e, n);
  return r !== -1 && r === n;
}, gn = (t) => {
  if (!t) return null;
  if (H(t)) return t;
  let e = t.length;
  if (!ut(e)) return null;
  const n = new Array(e);
  for (; e-- > 0; )
    n[e] = t[e];
  return n;
}, yn = /* @__PURE__ */ ((t) => (e) => t && e instanceof t)(typeof Uint8Array < "u" && _e(Uint8Array)), wn = (t, e) => {
  const r = (t && t[ce]).call(t);
  let s;
  for (; (s = r.next()) && !s.done; ) {
    const o = s.value;
    e.call(t, o[0], o[1]);
  }
}, bn = (t, e) => {
  let n;
  const r = [];
  for (; (n = t.exec(e)) !== null; )
    r.push(n);
  return r;
}, Rn = U("HTMLFormElement"), En = (t) => t.toLowerCase().replace(/[-_\s]([a-z\d])(\w*)/g, function(n, r, s) {
  return r.toUpperCase() + s;
}), He = (({ hasOwnProperty: t }) => (e, n) => t.call(e, n))(Object.prototype), Sn = U("RegExp"), pt = (t, e) => {
  const n = Object.getOwnPropertyDescriptors(t), r = {};
  G(n, (s, o) => {
    let i;
    (i = e(s, o, t)) !== !1 && (r[o] = i || s);
  }), Object.defineProperties(t, r);
}, On = (t) => {
  pt(t, (e, n) => {
    if (x(t) && ["arguments", "caller", "callee"].indexOf(n) !== -1)
      return !1;
    const r = t[n];
    if (x(r)) {
      if (e.enumerable = !1, "writable" in e) {
        e.writable = !1;
        return;
      }
      e.set || (e.set = () => {
        throw Error("Can not rewrite read-only method '" + n + "'");
      });
    }
  });
}, Tn = (t, e) => {
  const n = {}, r = (s) => {
    s.forEach((o) => {
      n[o] = !0;
    });
  };
  return H(t) ? r(t) : r(String(t).split(e)), n;
}, An = () => {
}, xn = (t, e) => t != null && Number.isFinite(t = +t) ? t : e;
function Cn(t) {
  return !!(t && x(t.append) && t[ct] === "FormData" && t[ce]);
}
const _n = (t) => {
  const e = new Array(10), n = (r, s) => {
    if (X(r)) {
      if (e.indexOf(r) >= 0)
        return;
      if (K(r))
        return r;
      if (!("toJSON" in r)) {
        e[s] = r;
        const o = H(r) ? [] : {};
        return G(r, (i, a) => {
          const f = n(i, s + 1);
          !M(f) && (o[a] = f);
        }), e[s] = void 0, o;
      }
    }
    return r;
  };
  return n(t, 0);
}, Pn = U("AsyncFunction"), Ln = (t) => t && (X(t) || x(t)) && x(t.then) && x(t.catch), ht = ((t, e) => t ? setImmediate : e ? ((n, r) => (B.addEventListener(
  "message",
  ({ source: s, data: o }) => {
    s === B && o === n && r.length && r.shift()();
  },
  !1
), (s) => {
  r.push(s), B.postMessage(n, "*");
}))(`axios@${Math.random()}`, []) : (n) => setTimeout(n))(typeof setImmediate == "function", x(B.postMessage)), Nn = typeof queueMicrotask < "u" ? queueMicrotask.bind(B) : typeof process < "u" && process.nextTick || ht, Un = (t) => t != null && x(t[ce]), c = {
  isArray: H,
  isArrayBuffer: lt,
  isBuffer: K,
  isFormData: rn,
  isArrayBufferView: Wt,
  isString: zt,
  isNumber: ut,
  isBoolean: Vt,
  isObject: X,
  isPlainObject: se,
  isEmptyObject: Kt,
  isReadableStream: on,
  isRequest: an,
  isResponse: cn,
  isHeaders: ln,
  isUndefined: M,
  isDate: Xt,
  isFile: Gt,
  isReactNativeBlob: Qt,
  isReactNative: Yt,
  isBlob: Zt,
  isRegExp: Sn,
  isFunction: x,
  isStream: tn,
  isURLSearchParams: sn,
  isTypedArray: yn,
  isFileList: en,
  forEach: G,
  merge: Se,
  extend: fn,
  trim: un,
  stripBOM: dn,
  inherits: pn,
  toFlatObject: hn,
  kindOf: le,
  kindOfTest: U,
  endsWith: mn,
  toArray: gn,
  forEachEntry: wn,
  matchAll: bn,
  isHTMLForm: Rn,
  hasOwnProperty: He,
  hasOwnProp: He,
  // an alias to avoid ESLint no-prototype-builtins detection
  reduceDescriptors: pt,
  freezeMethods: On,
  toObjectSet: Tn,
  toCamelCase: En,
  noop: An,
  toFiniteNumber: xn,
  findKey: ft,
  global: B,
  isContextDefined: dt,
  isSpecCompliantForm: Cn,
  toJSONObject: _n,
  isAsyncFn: Pn,
  isThenable: Ln,
  setImmediate: ht,
  asap: Nn,
  isIterable: Un
};
let g = class mt extends Error {
  static from(e, n, r, s, o, i) {
    const a = new mt(e.message, n || e.code, r, s, o);
    return a.cause = e, a.name = e.name, e.status != null && a.status == null && (a.status = e.status), i && Object.assign(a, i), a;
  }
  /**
   * Create an Error with the specified message, config, error code, request and response.
   *
   * @param {string} message The error message.
   * @param {string} [code] The error code (for example, 'ECONNABORTED').
   * @param {Object} [config] The config.
   * @param {Object} [request] The request.
   * @param {Object} [response] The response.
   *
   * @returns {Error} The created error.
   */
  constructor(e, n, r, s, o) {
    super(e), Object.defineProperty(this, "message", {
      value: e,
      enumerable: !0,
      writable: !0,
      configurable: !0
    }), this.name = "AxiosError", this.isAxiosError = !0, n && (this.code = n), r && (this.config = r), s && (this.request = s), o && (this.response = o, this.status = o.status);
  }
  toJSON() {
    return {
      // Standard
      message: this.message,
      name: this.name,
      // Microsoft
      description: this.description,
      number: this.number,
      // Mozilla
      fileName: this.fileName,
      lineNumber: this.lineNumber,
      columnNumber: this.columnNumber,
      stack: this.stack,
      // Axios
      config: c.toJSONObject(this.config),
      code: this.code,
      status: this.status
    };
  }
};
g.ERR_BAD_OPTION_VALUE = "ERR_BAD_OPTION_VALUE";
g.ERR_BAD_OPTION = "ERR_BAD_OPTION";
g.ECONNABORTED = "ECONNABORTED";
g.ETIMEDOUT = "ETIMEDOUT";
g.ERR_NETWORK = "ERR_NETWORK";
g.ERR_FR_TOO_MANY_REDIRECTS = "ERR_FR_TOO_MANY_REDIRECTS";
g.ERR_DEPRECATED = "ERR_DEPRECATED";
g.ERR_BAD_RESPONSE = "ERR_BAD_RESPONSE";
g.ERR_BAD_REQUEST = "ERR_BAD_REQUEST";
g.ERR_CANCELED = "ERR_CANCELED";
g.ERR_NOT_SUPPORT = "ERR_NOT_SUPPORT";
g.ERR_INVALID_URL = "ERR_INVALID_URL";
const Fn = null;
function Oe(t) {
  return c.isPlainObject(t) || c.isArray(t);
}
function gt(t) {
  return c.endsWith(t, "[]") ? t.slice(0, -2) : t;
}
function ye(t, e, n) {
  return t ? t.concat(e).map(function(s, o) {
    return s = gt(s), !n && o ? "[" + s + "]" : s;
  }).join(n ? "." : "") : e;
}
function jn(t) {
  return c.isArray(t) && !t.some(Oe);
}
const kn = c.toFlatObject(c, {}, null, function(e) {
  return /^is[A-Z]/.test(e);
});
function fe(t, e, n) {
  if (!c.isObject(t))
    throw new TypeError("target must be an object");
  e = e || new FormData(), n = c.toFlatObject(
    n,
    {
      metaTokens: !0,
      dots: !1,
      indexes: !1
    },
    !1,
    function(m, h) {
      return !c.isUndefined(h[m]);
    }
  );
  const r = n.metaTokens, s = n.visitor || l, o = n.dots, i = n.indexes, f = (n.Blob || typeof Blob < "u" && Blob) && c.isSpecCompliantForm(e);
  if (!c.isFunction(s))
    throw new TypeError("visitor must be a function");
  function u(d) {
    if (d === null) return "";
    if (c.isDate(d))
      return d.toISOString();
    if (c.isBoolean(d))
      return d.toString();
    if (!f && c.isBlob(d))
      throw new g("Blob is not supported. Use a Buffer instead.");
    return c.isArrayBuffer(d) || c.isTypedArray(d) ? f && typeof Blob == "function" ? new Blob([d]) : Buffer.from(d) : d;
  }
  function l(d, m, h) {
    let w = d;
    if (c.isReactNative(e) && c.isReactNativeBlob(d))
      return e.append(ye(h, m, o), u(d)), !1;
    if (d && !h && typeof d == "object") {
      if (c.endsWith(m, "{}"))
        m = r ? m : m.slice(0, -2), d = JSON.stringify(d);
      else if (c.isArray(d) && jn(d) || (c.isFileList(d) || c.endsWith(m, "[]")) && (w = c.toArray(d)))
        return m = gt(m), w.forEach(function(b, O) {
          !(c.isUndefined(b) || b === null) && e.append(
            // eslint-disable-next-line no-nested-ternary
            i === !0 ? ye([m], O, o) : i === null ? m : m + "[]",
            u(b)
          );
        }), !1;
    }
    return Oe(d) ? !0 : (e.append(ye(h, m, o), u(d)), !1);
  }
  const p = [], y = Object.assign(kn, {
    defaultVisitor: l,
    convertValue: u,
    isVisitable: Oe
  });
  function R(d, m) {
    if (!c.isUndefined(d)) {
      if (p.indexOf(d) !== -1)
        throw Error("Circular reference detected in " + m.join("."));
      p.push(d), c.forEach(d, function(w, S) {
        (!(c.isUndefined(w) || w === null) && s.call(e, w, c.isString(S) ? S.trim() : S, m, y)) === !0 && R(w, m ? m.concat(S) : [S]);
      }), p.pop();
    }
  }
  if (!c.isObject(t))
    throw new TypeError("data must be an object");
  return R(t), e;
}
function $e(t) {
  const e = {
    "!": "%21",
    "'": "%27",
    "(": "%28",
    ")": "%29",
    "~": "%7E",
    "%20": "+",
    "%00": "\0"
  };
  return encodeURIComponent(t).replace(/[!'()~]|%20|%00/g, function(r) {
    return e[r];
  });
}
function Pe(t, e) {
  this._pairs = [], t && fe(t, this, e);
}
const yt = Pe.prototype;
yt.append = function(e, n) {
  this._pairs.push([e, n]);
};
yt.toString = function(e) {
  const n = e ? function(r) {
    return e.call(this, r, $e);
  } : $e;
  return this._pairs.map(function(s) {
    return n(s[0]) + "=" + n(s[1]);
  }, "").join("&");
};
function Dn(t) {
  return encodeURIComponent(t).replace(/%3A/gi, ":").replace(/%24/g, "$").replace(/%2C/gi, ",").replace(/%20/g, "+");
}
function wt(t, e, n) {
  if (!e)
    return t;
  const r = n && n.encode || Dn, s = c.isFunction(n) ? {
    serialize: n
  } : n, o = s && s.serialize;
  let i;
  if (o ? i = o(e, s) : i = c.isURLSearchParams(e) ? e.toString() : new Pe(e, s).toString(r), i) {
    const a = t.indexOf("#");
    a !== -1 && (t = t.slice(0, a)), t += (t.indexOf("?") === -1 ? "?" : "&") + i;
  }
  return t;
}
class Je {
  constructor() {
    this.handlers = [];
  }
  /**
   * Add a new interceptor to the stack
   *
   * @param {Function} fulfilled The function to handle `then` for a `Promise`
   * @param {Function} rejected The function to handle `reject` for a `Promise`
   * @param {Object} options The options for the interceptor, synchronous and runWhen
   *
   * @return {Number} An ID used to remove interceptor later
   */
  use(e, n, r) {
    return this.handlers.push({
      fulfilled: e,
      rejected: n,
      synchronous: r ? r.synchronous : !1,
      runWhen: r ? r.runWhen : null
    }), this.handlers.length - 1;
  }
  /**
   * Remove an interceptor from the stack
   *
   * @param {Number} id The ID that was returned by `use`
   *
   * @returns {void}
   */
  eject(e) {
    this.handlers[e] && (this.handlers[e] = null);
  }
  /**
   * Clear all interceptors from the stack
   *
   * @returns {void}
   */
  clear() {
    this.handlers && (this.handlers = []);
  }
  /**
   * Iterate over all the registered interceptors
   *
   * This method is particularly useful for skipping over any
   * interceptors that may have become `null` calling `eject`.
   *
   * @param {Function} fn The function to call for each interceptor
   *
   * @returns {void}
   */
  forEach(e) {
    c.forEach(this.handlers, function(r) {
      r !== null && e(r);
    });
  }
}
const Le = {
  silentJSONParsing: !0,
  forcedJSONParsing: !0,
  clarifyTimeoutError: !1,
  legacyInterceptorReqResOrdering: !0
}, Bn = typeof URLSearchParams < "u" ? URLSearchParams : Pe, vn = typeof FormData < "u" ? FormData : null, In = typeof Blob < "u" ? Blob : null, qn = {
  isBrowser: !0,
  classes: {
    URLSearchParams: Bn,
    FormData: vn,
    Blob: In
  },
  protocols: ["http", "https", "file", "blob", "url", "data"]
}, Ne = typeof window < "u" && typeof document < "u", Te = typeof navigator == "object" && navigator || void 0, Mn = Ne && (!Te || ["ReactNative", "NativeScript", "NS"].indexOf(Te.product) < 0), Hn = typeof WorkerGlobalScope < "u" && // eslint-disable-next-line no-undef
self instanceof WorkerGlobalScope && typeof self.importScripts == "function", $n = Ne && window.location.href || "http://localhost", Jn = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  hasBrowserEnv: Ne,
  hasStandardBrowserEnv: Mn,
  hasStandardBrowserWebWorkerEnv: Hn,
  navigator: Te,
  origin: $n
}, Symbol.toStringTag, { value: "Module" })), T = {
  ...Jn,
  ...qn
};
function Wn(t, e) {
  return fe(t, new T.classes.URLSearchParams(), {
    visitor: function(n, r, s, o) {
      return T.isNode && c.isBuffer(n) ? (this.append(r, n.toString("base64")), !1) : o.defaultVisitor.apply(this, arguments);
    },
    ...e
  });
}
function zn(t) {
  return c.matchAll(/\w+|\[(\w*)]/g, t).map((e) => e[0] === "[]" ? "" : e[1] || e[0]);
}
function Vn(t) {
  const e = {}, n = Object.keys(t);
  let r;
  const s = n.length;
  let o;
  for (r = 0; r < s; r++)
    o = n[r], e[o] = t[o];
  return e;
}
function bt(t) {
  function e(n, r, s, o) {
    let i = n[o++];
    if (i === "__proto__") return !0;
    const a = Number.isFinite(+i), f = o >= n.length;
    return i = !i && c.isArray(s) ? s.length : i, f ? (c.hasOwnProp(s, i) ? s[i] = [s[i], r] : s[i] = r, !a) : ((!s[i] || !c.isObject(s[i])) && (s[i] = []), e(n, r, s[i], o) && c.isArray(s[i]) && (s[i] = Vn(s[i])), !a);
  }
  if (c.isFormData(t) && c.isFunction(t.entries)) {
    const n = {};
    return c.forEachEntry(t, (r, s) => {
      e(zn(r), s, n, 0);
    }), n;
  }
  return null;
}
function Kn(t, e, n) {
  if (c.isString(t))
    try {
      return (e || JSON.parse)(t), c.trim(t);
    } catch (r) {
      if (r.name !== "SyntaxError")
        throw r;
    }
  return (n || JSON.stringify)(t);
}
const Q = {
  transitional: Le,
  adapter: ["xhr", "http", "fetch"],
  transformRequest: [
    function(e, n) {
      const r = n.getContentType() || "", s = r.indexOf("application/json") > -1, o = c.isObject(e);
      if (o && c.isHTMLForm(e) && (e = new FormData(e)), c.isFormData(e))
        return s ? JSON.stringify(bt(e)) : e;
      if (c.isArrayBuffer(e) || c.isBuffer(e) || c.isStream(e) || c.isFile(e) || c.isBlob(e) || c.isReadableStream(e))
        return e;
      if (c.isArrayBufferView(e))
        return e.buffer;
      if (c.isURLSearchParams(e))
        return n.setContentType("application/x-www-form-urlencoded;charset=utf-8", !1), e.toString();
      let a;
      if (o) {
        if (r.indexOf("application/x-www-form-urlencoded") > -1)
          return Wn(e, this.formSerializer).toString();
        if ((a = c.isFileList(e)) || r.indexOf("multipart/form-data") > -1) {
          const f = this.env && this.env.FormData;
          return fe(
            a ? { "files[]": e } : e,
            f && new f(),
            this.formSerializer
          );
        }
      }
      return o || s ? (n.setContentType("application/json", !1), Kn(e)) : e;
    }
  ],
  transformResponse: [
    function(e) {
      const n = this.transitional || Q.transitional, r = n && n.forcedJSONParsing, s = this.responseType === "json";
      if (c.isResponse(e) || c.isReadableStream(e))
        return e;
      if (e && c.isString(e) && (r && !this.responseType || s)) {
        const i = !(n && n.silentJSONParsing) && s;
        try {
          return JSON.parse(e, this.parseReviver);
        } catch (a) {
          if (i)
            throw a.name === "SyntaxError" ? g.from(a, g.ERR_BAD_RESPONSE, this, null, this.response) : a;
        }
      }
      return e;
    }
  ],
  /**
   * A timeout in milliseconds to abort a request. If set to 0 (default) a
   * timeout is not created.
   */
  timeout: 0,
  xsrfCookieName: "XSRF-TOKEN",
  xsrfHeaderName: "X-XSRF-TOKEN",
  maxContentLength: -1,
  maxBodyLength: -1,
  env: {
    FormData: T.classes.FormData,
    Blob: T.classes.Blob
  },
  validateStatus: function(e) {
    return e >= 200 && e < 300;
  },
  headers: {
    common: {
      Accept: "application/json, text/plain, */*",
      "Content-Type": void 0
    }
  }
};
c.forEach(["delete", "get", "head", "post", "put", "patch"], (t) => {
  Q.headers[t] = {};
});
const Xn = c.toObjectSet([
  "age",
  "authorization",
  "content-length",
  "content-type",
  "etag",
  "expires",
  "from",
  "host",
  "if-modified-since",
  "if-unmodified-since",
  "last-modified",
  "location",
  "max-forwards",
  "proxy-authorization",
  "referer",
  "retry-after",
  "user-agent"
]), Gn = (t) => {
  const e = {};
  let n, r, s;
  return t && t.split(`
`).forEach(function(i) {
    s = i.indexOf(":"), n = i.substring(0, s).trim().toLowerCase(), r = i.substring(s + 1).trim(), !(!n || e[n] && Xn[n]) && (n === "set-cookie" ? e[n] ? e[n].push(r) : e[n] = [r] : e[n] = e[n] ? e[n] + ", " + r : r);
  }), e;
}, We = Symbol("internals");
function J(t) {
  return t && String(t).trim().toLowerCase();
}
function oe(t) {
  return t === !1 || t == null ? t : c.isArray(t) ? t.map(oe) : String(t);
}
function Qn(t) {
  const e = /* @__PURE__ */ Object.create(null), n = /([^\s,;=]+)\s*(?:=\s*([^,;]+))?/g;
  let r;
  for (; r = n.exec(t); )
    e[r[1]] = r[2];
  return e;
}
const Yn = (t) => /^[-_a-zA-Z0-9^`|~,!#$%&'*+.]+$/.test(t.trim());
function we(t, e, n, r, s) {
  if (c.isFunction(r))
    return r.call(this, e, n);
  if (s && (e = n), !!c.isString(e)) {
    if (c.isString(r))
      return e.indexOf(r) !== -1;
    if (c.isRegExp(r))
      return r.test(e);
  }
}
function Zn(t) {
  return t.trim().toLowerCase().replace(/([a-z\d])(\w*)/g, (e, n, r) => n.toUpperCase() + r);
}
function er(t, e) {
  const n = c.toCamelCase(" " + e);
  ["get", "set", "has"].forEach((r) => {
    Object.defineProperty(t, r + n, {
      value: function(s, o, i) {
        return this[r].call(this, e, s, o, i);
      },
      configurable: !0
    });
  });
}
let C = class {
  constructor(e) {
    e && this.set(e);
  }
  set(e, n, r) {
    const s = this;
    function o(a, f, u) {
      const l = J(f);
      if (!l)
        throw new Error("header name must be a non-empty string");
      const p = c.findKey(s, l);
      (!p || s[p] === void 0 || u === !0 || u === void 0 && s[p] !== !1) && (s[p || f] = oe(a));
    }
    const i = (a, f) => c.forEach(a, (u, l) => o(u, l, f));
    if (c.isPlainObject(e) || e instanceof this.constructor)
      i(e, n);
    else if (c.isString(e) && (e = e.trim()) && !Yn(e))
      i(Gn(e), n);
    else if (c.isObject(e) && c.isIterable(e)) {
      let a = {}, f, u;
      for (const l of e) {
        if (!c.isArray(l))
          throw TypeError("Object iterator must return a key-value pair");
        a[u = l[0]] = (f = a[u]) ? c.isArray(f) ? [...f, l[1]] : [f, l[1]] : l[1];
      }
      i(a, n);
    } else
      e != null && o(n, e, r);
    return this;
  }
  get(e, n) {
    if (e = J(e), e) {
      const r = c.findKey(this, e);
      if (r) {
        const s = this[r];
        if (!n)
          return s;
        if (n === !0)
          return Qn(s);
        if (c.isFunction(n))
          return n.call(this, s, r);
        if (c.isRegExp(n))
          return n.exec(s);
        throw new TypeError("parser must be boolean|regexp|function");
      }
    }
  }
  has(e, n) {
    if (e = J(e), e) {
      const r = c.findKey(this, e);
      return !!(r && this[r] !== void 0 && (!n || we(this, this[r], r, n)));
    }
    return !1;
  }
  delete(e, n) {
    const r = this;
    let s = !1;
    function o(i) {
      if (i = J(i), i) {
        const a = c.findKey(r, i);
        a && (!n || we(r, r[a], a, n)) && (delete r[a], s = !0);
      }
    }
    return c.isArray(e) ? e.forEach(o) : o(e), s;
  }
  clear(e) {
    const n = Object.keys(this);
    let r = n.length, s = !1;
    for (; r--; ) {
      const o = n[r];
      (!e || we(this, this[o], o, e, !0)) && (delete this[o], s = !0);
    }
    return s;
  }
  normalize(e) {
    const n = this, r = {};
    return c.forEach(this, (s, o) => {
      const i = c.findKey(r, o);
      if (i) {
        n[i] = oe(s), delete n[o];
        return;
      }
      const a = e ? Zn(o) : String(o).trim();
      a !== o && delete n[o], n[a] = oe(s), r[a] = !0;
    }), this;
  }
  concat(...e) {
    return this.constructor.concat(this, ...e);
  }
  toJSON(e) {
    const n = /* @__PURE__ */ Object.create(null);
    return c.forEach(this, (r, s) => {
      r != null && r !== !1 && (n[s] = e && c.isArray(r) ? r.join(", ") : r);
    }), n;
  }
  [Symbol.iterator]() {
    return Object.entries(this.toJSON())[Symbol.iterator]();
  }
  toString() {
    return Object.entries(this.toJSON()).map(([e, n]) => e + ": " + n).join(`
`);
  }
  getSetCookie() {
    return this.get("set-cookie") || [];
  }
  get [Symbol.toStringTag]() {
    return "AxiosHeaders";
  }
  static from(e) {
    return e instanceof this ? e : new this(e);
  }
  static concat(e, ...n) {
    const r = new this(e);
    return n.forEach((s) => r.set(s)), r;
  }
  static accessor(e) {
    const r = (this[We] = this[We] = {
      accessors: {}
    }).accessors, s = this.prototype;
    function o(i) {
      const a = J(i);
      r[a] || (er(s, i), r[a] = !0);
    }
    return c.isArray(e) ? e.forEach(o) : o(e), this;
  }
};
C.accessor([
  "Content-Type",
  "Content-Length",
  "Accept",
  "Accept-Encoding",
  "User-Agent",
  "Authorization"
]);
c.reduceDescriptors(C.prototype, ({ value: t }, e) => {
  let n = e[0].toUpperCase() + e.slice(1);
  return {
    get: () => t,
    set(r) {
      this[n] = r;
    }
  };
});
c.freezeMethods(C);
function be(t, e) {
  const n = this || Q, r = e || n, s = C.from(r.headers);
  let o = r.data;
  return c.forEach(t, function(a) {
    o = a.call(n, o, s.normalize(), e ? e.status : void 0);
  }), s.normalize(), o;
}
function Rt(t) {
  return !!(t && t.__CANCEL__);
}
let Y = class extends g {
  /**
   * A `CanceledError` is an object that is thrown when an operation is canceled.
   *
   * @param {string=} message The message.
   * @param {Object=} config The config.
   * @param {Object=} request The request.
   *
   * @returns {CanceledError} The created error.
   */
  constructor(e, n, r) {
    super(e ?? "canceled", g.ERR_CANCELED, n, r), this.name = "CanceledError", this.__CANCEL__ = !0;
  }
};
function Et(t, e, n) {
  const r = n.config.validateStatus;
  !n.status || !r || r(n.status) ? t(n) : e(
    new g(
      "Request failed with status code " + n.status,
      [g.ERR_BAD_REQUEST, g.ERR_BAD_RESPONSE][Math.floor(n.status / 100) - 4],
      n.config,
      n.request,
      n
    )
  );
}
function tr(t) {
  const e = /^([-+\w]{1,25})(:?\/\/|:)/.exec(t);
  return e && e[1] || "";
}
function nr(t, e) {
  t = t || 10;
  const n = new Array(t), r = new Array(t);
  let s = 0, o = 0, i;
  return e = e !== void 0 ? e : 1e3, function(f) {
    const u = Date.now(), l = r[o];
    i || (i = u), n[s] = f, r[s] = u;
    let p = o, y = 0;
    for (; p !== s; )
      y += n[p++], p = p % t;
    if (s = (s + 1) % t, s === o && (o = (o + 1) % t), u - i < e)
      return;
    const R = l && u - l;
    return R ? Math.round(y * 1e3 / R) : void 0;
  };
}
function rr(t, e) {
  let n = 0, r = 1e3 / e, s, o;
  const i = (u, l = Date.now()) => {
    n = l, s = null, o && (clearTimeout(o), o = null), t(...u);
  };
  return [(...u) => {
    const l = Date.now(), p = l - n;
    p >= r ? i(u, l) : (s = u, o || (o = setTimeout(() => {
      o = null, i(s);
    }, r - p)));
  }, () => s && i(s)];
}
const ae = (t, e, n = 3) => {
  let r = 0;
  const s = nr(50, 250);
  return rr((o) => {
    const i = o.loaded, a = o.lengthComputable ? o.total : void 0, f = i - r, u = s(f), l = i <= a;
    r = i;
    const p = {
      loaded: i,
      total: a,
      progress: a ? i / a : void 0,
      bytes: f,
      rate: u || void 0,
      estimated: u && a && l ? (a - i) / u : void 0,
      event: o,
      lengthComputable: a != null,
      [e ? "download" : "upload"]: !0
    };
    t(p);
  }, n);
}, ze = (t, e) => {
  const n = t != null;
  return [
    (r) => e[0]({
      lengthComputable: n,
      total: t,
      loaded: r
    }),
    e[1]
  ];
}, Ve = (t) => (...e) => c.asap(() => t(...e)), sr = T.hasStandardBrowserEnv ? /* @__PURE__ */ ((t, e) => (n) => (n = new URL(n, T.origin), t.protocol === n.protocol && t.host === n.host && (e || t.port === n.port)))(
  new URL(T.origin),
  T.navigator && /(msie|trident)/i.test(T.navigator.userAgent)
) : () => !0, or = T.hasStandardBrowserEnv ? (
  // Standard browser envs support document.cookie
  {
    write(t, e, n, r, s, o, i) {
      if (typeof document > "u") return;
      const a = [`${t}=${encodeURIComponent(e)}`];
      c.isNumber(n) && a.push(`expires=${new Date(n).toUTCString()}`), c.isString(r) && a.push(`path=${r}`), c.isString(s) && a.push(`domain=${s}`), o === !0 && a.push("secure"), c.isString(i) && a.push(`SameSite=${i}`), document.cookie = a.join("; ");
    },
    read(t) {
      if (typeof document > "u") return null;
      const e = document.cookie.match(new RegExp("(?:^|; )" + t + "=([^;]*)"));
      return e ? decodeURIComponent(e[1]) : null;
    },
    remove(t) {
      this.write(t, "", Date.now() - 864e5, "/");
    }
  }
) : (
  // Non-standard browser env (web workers, react-native) lack needed support.
  {
    write() {
    },
    read() {
      return null;
    },
    remove() {
    }
  }
);
function ir(t) {
  return typeof t != "string" ? !1 : /^([a-z][a-z\d+\-.]*:)?\/\//i.test(t);
}
function ar(t, e) {
  return e ? t.replace(/\/?\/$/, "") + "/" + e.replace(/^\/+/, "") : t;
}
function St(t, e, n) {
  let r = !ir(e);
  return t && (r || n == !1) ? ar(t, e) : e;
}
const Ke = (t) => t instanceof C ? { ...t } : t;
function I(t, e) {
  e = e || {};
  const n = {};
  function r(u, l, p, y) {
    return c.isPlainObject(u) && c.isPlainObject(l) ? c.merge.call({ caseless: y }, u, l) : c.isPlainObject(l) ? c.merge({}, l) : c.isArray(l) ? l.slice() : l;
  }
  function s(u, l, p, y) {
    if (c.isUndefined(l)) {
      if (!c.isUndefined(u))
        return r(void 0, u, p, y);
    } else return r(u, l, p, y);
  }
  function o(u, l) {
    if (!c.isUndefined(l))
      return r(void 0, l);
  }
  function i(u, l) {
    if (c.isUndefined(l)) {
      if (!c.isUndefined(u))
        return r(void 0, u);
    } else return r(void 0, l);
  }
  function a(u, l, p) {
    if (p in e)
      return r(u, l);
    if (p in t)
      return r(void 0, u);
  }
  const f = {
    url: o,
    method: o,
    data: o,
    baseURL: i,
    transformRequest: i,
    transformResponse: i,
    paramsSerializer: i,
    timeout: i,
    timeoutMessage: i,
    withCredentials: i,
    withXSRFToken: i,
    adapter: i,
    responseType: i,
    xsrfCookieName: i,
    xsrfHeaderName: i,
    onUploadProgress: i,
    onDownloadProgress: i,
    decompress: i,
    maxContentLength: i,
    maxBodyLength: i,
    beforeRedirect: i,
    transport: i,
    httpAgent: i,
    httpsAgent: i,
    cancelToken: i,
    socketPath: i,
    responseEncoding: i,
    validateStatus: a,
    headers: (u, l, p) => s(Ke(u), Ke(l), p, !0)
  };
  return c.forEach(Object.keys({ ...t, ...e }), function(l) {
    if (l === "__proto__" || l === "constructor" || l === "prototype") return;
    const p = c.hasOwnProp(f, l) ? f[l] : s, y = p(t[l], e[l], l);
    c.isUndefined(y) && p !== a || (n[l] = y);
  }), n;
}
const Ot = (t) => {
  const e = I({}, t);
  let { data: n, withXSRFToken: r, xsrfHeaderName: s, xsrfCookieName: o, headers: i, auth: a } = e;
  if (e.headers = i = C.from(i), e.url = wt(
    St(e.baseURL, e.url, e.allowAbsoluteUrls),
    t.params,
    t.paramsSerializer
  ), a && i.set(
    "Authorization",
    "Basic " + btoa(
      (a.username || "") + ":" + (a.password ? unescape(encodeURIComponent(a.password)) : "")
    )
  ), c.isFormData(n)) {
    if (T.hasStandardBrowserEnv || T.hasStandardBrowserWebWorkerEnv)
      i.setContentType(void 0);
    else if (c.isFunction(n.getHeaders)) {
      const f = n.getHeaders(), u = ["content-type", "content-length"];
      Object.entries(f).forEach(([l, p]) => {
        u.includes(l.toLowerCase()) && i.set(l, p);
      });
    }
  }
  if (T.hasStandardBrowserEnv && (r && c.isFunction(r) && (r = r(e)), r || r !== !1 && sr(e.url))) {
    const f = s && o && or.read(o);
    f && i.set(s, f);
  }
  return e;
}, cr = typeof XMLHttpRequest < "u", lr = cr && function(t) {
  return new Promise(function(n, r) {
    const s = Ot(t);
    let o = s.data;
    const i = C.from(s.headers).normalize();
    let { responseType: a, onUploadProgress: f, onDownloadProgress: u } = s, l, p, y, R, d;
    function m() {
      R && R(), d && d(), s.cancelToken && s.cancelToken.unsubscribe(l), s.signal && s.signal.removeEventListener("abort", l);
    }
    let h = new XMLHttpRequest();
    h.open(s.method.toUpperCase(), s.url, !0), h.timeout = s.timeout;
    function w() {
      if (!h)
        return;
      const b = C.from(
        "getAllResponseHeaders" in h && h.getAllResponseHeaders()
      ), P = {
        data: !a || a === "text" || a === "json" ? h.responseText : h.response,
        status: h.status,
        statusText: h.statusText,
        headers: b,
        config: t,
        request: h
      };
      Et(
        function(L) {
          n(L), m();
        },
        function(L) {
          r(L), m();
        },
        P
      ), h = null;
    }
    "onloadend" in h ? h.onloadend = w : h.onreadystatechange = function() {
      !h || h.readyState !== 4 || h.status === 0 && !(h.responseURL && h.responseURL.indexOf("file:") === 0) || setTimeout(w);
    }, h.onabort = function() {
      h && (r(new g("Request aborted", g.ECONNABORTED, t, h)), h = null);
    }, h.onerror = function(O) {
      const P = O && O.message ? O.message : "Network Error", k = new g(P, g.ERR_NETWORK, t, h);
      k.event = O || null, r(k), h = null;
    }, h.ontimeout = function() {
      let O = s.timeout ? "timeout of " + s.timeout + "ms exceeded" : "timeout exceeded";
      const P = s.transitional || Le;
      s.timeoutErrorMessage && (O = s.timeoutErrorMessage), r(
        new g(
          O,
          P.clarifyTimeoutError ? g.ETIMEDOUT : g.ECONNABORTED,
          t,
          h
        )
      ), h = null;
    }, o === void 0 && i.setContentType(null), "setRequestHeader" in h && c.forEach(i.toJSON(), function(O, P) {
      h.setRequestHeader(P, O);
    }), c.isUndefined(s.withCredentials) || (h.withCredentials = !!s.withCredentials), a && a !== "json" && (h.responseType = s.responseType), u && ([y, d] = ae(u, !0), h.addEventListener("progress", y)), f && h.upload && ([p, R] = ae(f), h.upload.addEventListener("progress", p), h.upload.addEventListener("loadend", R)), (s.cancelToken || s.signal) && (l = (b) => {
      h && (r(!b || b.type ? new Y(null, t, h) : b), h.abort(), h = null);
    }, s.cancelToken && s.cancelToken.subscribe(l), s.signal && (s.signal.aborted ? l() : s.signal.addEventListener("abort", l)));
    const S = tr(s.url);
    if (S && T.protocols.indexOf(S) === -1) {
      r(
        new g(
          "Unsupported protocol " + S + ":",
          g.ERR_BAD_REQUEST,
          t
        )
      );
      return;
    }
    h.send(o || null);
  });
}, ur = (t, e) => {
  const { length: n } = t = t ? t.filter(Boolean) : [];
  if (e || n) {
    let r = new AbortController(), s;
    const o = function(u) {
      if (!s) {
        s = !0, a();
        const l = u instanceof Error ? u : this.reason;
        r.abort(
          l instanceof g ? l : new Y(l instanceof Error ? l.message : l)
        );
      }
    };
    let i = e && setTimeout(() => {
      i = null, o(new g(`timeout of ${e}ms exceeded`, g.ETIMEDOUT));
    }, e);
    const a = () => {
      t && (i && clearTimeout(i), i = null, t.forEach((u) => {
        u.unsubscribe ? u.unsubscribe(o) : u.removeEventListener("abort", o);
      }), t = null);
    };
    t.forEach((u) => u.addEventListener("abort", o));
    const { signal: f } = r;
    return f.unsubscribe = () => c.asap(a), f;
  }
}, fr = function* (t, e) {
  let n = t.byteLength;
  if (n < e) {
    yield t;
    return;
  }
  let r = 0, s;
  for (; r < n; )
    s = r + e, yield t.slice(r, s), r = s;
}, dr = async function* (t, e) {
  for await (const n of pr(t))
    yield* fr(n, e);
}, pr = async function* (t) {
  if (t[Symbol.asyncIterator]) {
    yield* t;
    return;
  }
  const e = t.getReader();
  try {
    for (; ; ) {
      const { done: n, value: r } = await e.read();
      if (n)
        break;
      yield r;
    }
  } finally {
    await e.cancel();
  }
}, Xe = (t, e, n, r) => {
  const s = dr(t, e);
  let o = 0, i, a = (f) => {
    i || (i = !0, r && r(f));
  };
  return new ReadableStream(
    {
      async pull(f) {
        try {
          const { done: u, value: l } = await s.next();
          if (u) {
            a(), f.close();
            return;
          }
          let p = l.byteLength;
          if (n) {
            let y = o += p;
            n(y);
          }
          f.enqueue(new Uint8Array(l));
        } catch (u) {
          throw a(u), u;
        }
      },
      cancel(f) {
        return a(f), s.return();
      }
    },
    {
      highWaterMark: 2
    }
  );
}, Ge = 64 * 1024, { isFunction: ne } = c, hr = (({ Request: t, Response: e }) => ({
  Request: t,
  Response: e
}))(c.global), { ReadableStream: Qe, TextEncoder: Ye } = c.global, Ze = (t, ...e) => {
  try {
    return !!t(...e);
  } catch {
    return !1;
  }
}, mr = (t) => {
  t = c.merge.call(
    {
      skipUndefined: !0
    },
    hr,
    t
  );
  const { fetch: e, Request: n, Response: r } = t, s = e ? ne(e) : typeof fetch == "function", o = ne(n), i = ne(r);
  if (!s)
    return !1;
  const a = s && ne(Qe), f = s && (typeof Ye == "function" ? /* @__PURE__ */ ((d) => (m) => d.encode(m))(new Ye()) : async (d) => new Uint8Array(await new n(d).arrayBuffer())), u = o && a && Ze(() => {
    let d = !1;
    const m = new n(T.origin, {
      body: new Qe(),
      method: "POST",
      get duplex() {
        return d = !0, "half";
      }
    }).headers.has("Content-Type");
    return d && !m;
  }), l = i && a && Ze(() => c.isReadableStream(new r("").body)), p = {
    stream: l && ((d) => d.body)
  };
  s && ["text", "arrayBuffer", "blob", "formData", "stream"].forEach((d) => {
    !p[d] && (p[d] = (m, h) => {
      let w = m && m[d];
      if (w)
        return w.call(m);
      throw new g(
        `Response type '${d}' is not supported`,
        g.ERR_NOT_SUPPORT,
        h
      );
    });
  });
  const y = async (d) => {
    if (d == null)
      return 0;
    if (c.isBlob(d))
      return d.size;
    if (c.isSpecCompliantForm(d))
      return (await new n(T.origin, {
        method: "POST",
        body: d
      }).arrayBuffer()).byteLength;
    if (c.isArrayBufferView(d) || c.isArrayBuffer(d))
      return d.byteLength;
    if (c.isURLSearchParams(d) && (d = d + ""), c.isString(d))
      return (await f(d)).byteLength;
  }, R = async (d, m) => {
    const h = c.toFiniteNumber(d.getContentLength());
    return h ?? y(m);
  };
  return async (d) => {
    let {
      url: m,
      method: h,
      data: w,
      signal: S,
      cancelToken: b,
      timeout: O,
      onDownloadProgress: P,
      onUploadProgress: k,
      responseType: L,
      headers: pe,
      withCredentials: Z = "same-origin",
      fetchOptions: je
    } = Ot(d), ke = e || fetch;
    L = L ? (L + "").toLowerCase() : "text";
    let ee = ur(
      [S, b && b.toAbortSignal()],
      O
    ), $ = null;
    const D = ee && ee.unsubscribe && (() => {
      ee.unsubscribe();
    });
    let De;
    try {
      if (k && u && h !== "get" && h !== "head" && (De = await R(pe, w)) !== 0) {
        let j = new n(m, {
          method: "POST",
          body: w,
          duplex: "half"
        }), q;
        if (c.isFormData(w) && (q = j.headers.get("content-type")) && pe.setContentType(q), j.body) {
          const [he, te] = ze(
            De,
            ae(Ve(k))
          );
          w = Xe(j.body, Ge, he, te);
        }
      }
      c.isString(Z) || (Z = Z ? "include" : "omit");
      const A = o && "credentials" in n.prototype, Be = {
        ...je,
        signal: ee,
        method: h.toUpperCase(),
        headers: pe.normalize().toJSON(),
        body: w,
        duplex: "half",
        credentials: A ? Z : void 0
      };
      $ = o && new n(m, Be);
      let F = await (o ? ke($, je) : ke(m, Be));
      const ve = l && (L === "stream" || L === "response");
      if (l && (P || ve && D)) {
        const j = {};
        ["status", "statusText", "headers"].forEach((Ie) => {
          j[Ie] = F[Ie];
        });
        const q = c.toFiniteNumber(F.headers.get("content-length")), [he, te] = P && ze(
          q,
          ae(Ve(P), !0)
        ) || [];
        F = new r(
          Xe(F.body, Ge, he, () => {
            te && te(), D && D();
          }),
          j
        );
      }
      L = L || "text";
      let jt = await p[c.findKey(p, L) || "text"](
        F,
        d
      );
      return !ve && D && D(), await new Promise((j, q) => {
        Et(j, q, {
          data: jt,
          headers: C.from(F.headers),
          status: F.status,
          statusText: F.statusText,
          config: d,
          request: $
        });
      });
    } catch (A) {
      throw D && D(), A && A.name === "TypeError" && /Load failed|fetch/i.test(A.message) ? Object.assign(
        new g(
          "Network Error",
          g.ERR_NETWORK,
          d,
          $,
          A && A.response
        ),
        {
          cause: A.cause || A
        }
      ) : g.from(A, A && A.code, d, $, A && A.response);
    }
  };
}, gr = /* @__PURE__ */ new Map(), Tt = (t) => {
  let e = t && t.env || {};
  const { fetch: n, Request: r, Response: s } = e, o = [r, s, n];
  let i = o.length, a = i, f, u, l = gr;
  for (; a--; )
    f = o[a], u = l.get(f), u === void 0 && l.set(f, u = a ? /* @__PURE__ */ new Map() : mr(e)), l = u;
  return u;
};
Tt();
const Ue = {
  http: Fn,
  xhr: lr,
  fetch: {
    get: Tt
  }
};
c.forEach(Ue, (t, e) => {
  if (t) {
    try {
      Object.defineProperty(t, "name", { value: e });
    } catch {
    }
    Object.defineProperty(t, "adapterName", { value: e });
  }
});
const et = (t) => `- ${t}`, yr = (t) => c.isFunction(t) || t === null || t === !1;
function wr(t, e) {
  t = c.isArray(t) ? t : [t];
  const { length: n } = t;
  let r, s;
  const o = {};
  for (let i = 0; i < n; i++) {
    r = t[i];
    let a;
    if (s = r, !yr(r) && (s = Ue[(a = String(r)).toLowerCase()], s === void 0))
      throw new g(`Unknown adapter '${a}'`);
    if (s && (c.isFunction(s) || (s = s.get(e))))
      break;
    o[a || "#" + i] = s;
  }
  if (!s) {
    const i = Object.entries(o).map(
      ([f, u]) => `adapter ${f} ` + (u === !1 ? "is not supported by the environment" : "is not available in the build")
    );
    let a = n ? i.length > 1 ? `since :
` + i.map(et).join(`
`) : " " + et(i[0]) : "as no adapter specified";
    throw new g(
      "There is no suitable adapter to dispatch the request " + a,
      "ERR_NOT_SUPPORT"
    );
  }
  return s;
}
const At = {
  /**
   * Resolve an adapter from a list of adapter names or functions.
   * @type {Function}
   */
  getAdapter: wr,
  /**
   * Exposes all known adapters
   * @type {Object<string, Function|Object>}
   */
  adapters: Ue
};
function Re(t) {
  if (t.cancelToken && t.cancelToken.throwIfRequested(), t.signal && t.signal.aborted)
    throw new Y(null, t);
}
function tt(t) {
  return Re(t), t.headers = C.from(t.headers), t.data = be.call(t, t.transformRequest), ["post", "put", "patch"].indexOf(t.method) !== -1 && t.headers.setContentType("application/x-www-form-urlencoded", !1), At.getAdapter(t.adapter || Q.adapter, t)(t).then(
    function(r) {
      return Re(t), r.data = be.call(t, t.transformResponse, r), r.headers = C.from(r.headers), r;
    },
    function(r) {
      return Rt(r) || (Re(t), r && r.response && (r.response.data = be.call(
        t,
        t.transformResponse,
        r.response
      ), r.response.headers = C.from(r.response.headers))), Promise.reject(r);
    }
  );
}
const xt = "1.13.6", de = {};
["object", "boolean", "number", "function", "string", "symbol"].forEach((t, e) => {
  de[t] = function(r) {
    return typeof r === t || "a" + (e < 1 ? "n " : " ") + t;
  };
});
const nt = {};
de.transitional = function(e, n, r) {
  function s(o, i) {
    return "[Axios v" + xt + "] Transitional option '" + o + "'" + i + (r ? ". " + r : "");
  }
  return (o, i, a) => {
    if (e === !1)
      throw new g(
        s(i, " has been removed" + (n ? " in " + n : "")),
        g.ERR_DEPRECATED
      );
    return n && !nt[i] && (nt[i] = !0, console.warn(
      s(
        i,
        " has been deprecated since v" + n + " and will be removed in the near future"
      )
    )), e ? e(o, i, a) : !0;
  };
};
de.spelling = function(e) {
  return (n, r) => (console.warn(`${r} is likely a misspelling of ${e}`), !0);
};
function br(t, e, n) {
  if (typeof t != "object")
    throw new g("options must be an object", g.ERR_BAD_OPTION_VALUE);
  const r = Object.keys(t);
  let s = r.length;
  for (; s-- > 0; ) {
    const o = r[s], i = e[o];
    if (i) {
      const a = t[o], f = a === void 0 || i(a, o, t);
      if (f !== !0)
        throw new g(
          "option " + o + " must be " + f,
          g.ERR_BAD_OPTION_VALUE
        );
      continue;
    }
    if (n !== !0)
      throw new g("Unknown option " + o, g.ERR_BAD_OPTION);
  }
}
const ie = {
  assertOptions: br,
  validators: de
}, N = ie.validators;
let v = class {
  constructor(e) {
    this.defaults = e || {}, this.interceptors = {
      request: new Je(),
      response: new Je()
    };
  }
  /**
   * Dispatch a request
   *
   * @param {String|Object} configOrUrl The config specific for this request (merged with this.defaults)
   * @param {?Object} config
   *
   * @returns {Promise} The Promise to be fulfilled
   */
  async request(e, n) {
    try {
      return await this._request(e, n);
    } catch (r) {
      if (r instanceof Error) {
        let s = {};
        Error.captureStackTrace ? Error.captureStackTrace(s) : s = new Error();
        const o = s.stack ? s.stack.replace(/^.+\n/, "") : "";
        try {
          r.stack ? o && !String(r.stack).endsWith(o.replace(/^.+\n.+\n/, "")) && (r.stack += `
` + o) : r.stack = o;
        } catch {
        }
      }
      throw r;
    }
  }
  _request(e, n) {
    typeof e == "string" ? (n = n || {}, n.url = e) : n = e || {}, n = I(this.defaults, n);
    const { transitional: r, paramsSerializer: s, headers: o } = n;
    r !== void 0 && ie.assertOptions(
      r,
      {
        silentJSONParsing: N.transitional(N.boolean),
        forcedJSONParsing: N.transitional(N.boolean),
        clarifyTimeoutError: N.transitional(N.boolean),
        legacyInterceptorReqResOrdering: N.transitional(N.boolean)
      },
      !1
    ), s != null && (c.isFunction(s) ? n.paramsSerializer = {
      serialize: s
    } : ie.assertOptions(
      s,
      {
        encode: N.function,
        serialize: N.function
      },
      !0
    )), n.allowAbsoluteUrls !== void 0 || (this.defaults.allowAbsoluteUrls !== void 0 ? n.allowAbsoluteUrls = this.defaults.allowAbsoluteUrls : n.allowAbsoluteUrls = !0), ie.assertOptions(
      n,
      {
        baseUrl: N.spelling("baseURL"),
        withXsrfToken: N.spelling("withXSRFToken")
      },
      !0
    ), n.method = (n.method || this.defaults.method || "get").toLowerCase();
    let i = o && c.merge(o.common, o[n.method]);
    o && c.forEach(["delete", "get", "head", "post", "put", "patch", "common"], (d) => {
      delete o[d];
    }), n.headers = C.concat(i, o);
    const a = [];
    let f = !0;
    this.interceptors.request.forEach(function(m) {
      if (typeof m.runWhen == "function" && m.runWhen(n) === !1)
        return;
      f = f && m.synchronous;
      const h = n.transitional || Le;
      h && h.legacyInterceptorReqResOrdering ? a.unshift(m.fulfilled, m.rejected) : a.push(m.fulfilled, m.rejected);
    });
    const u = [];
    this.interceptors.response.forEach(function(m) {
      u.push(m.fulfilled, m.rejected);
    });
    let l, p = 0, y;
    if (!f) {
      const d = [tt.bind(this), void 0];
      for (d.unshift(...a), d.push(...u), y = d.length, l = Promise.resolve(n); p < y; )
        l = l.then(d[p++], d[p++]);
      return l;
    }
    y = a.length;
    let R = n;
    for (; p < y; ) {
      const d = a[p++], m = a[p++];
      try {
        R = d(R);
      } catch (h) {
        m.call(this, h);
        break;
      }
    }
    try {
      l = tt.call(this, R);
    } catch (d) {
      return Promise.reject(d);
    }
    for (p = 0, y = u.length; p < y; )
      l = l.then(u[p++], u[p++]);
    return l;
  }
  getUri(e) {
    e = I(this.defaults, e);
    const n = St(e.baseURL, e.url, e.allowAbsoluteUrls);
    return wt(n, e.params, e.paramsSerializer);
  }
};
c.forEach(["delete", "get", "head", "options"], function(e) {
  v.prototype[e] = function(n, r) {
    return this.request(
      I(r || {}, {
        method: e,
        url: n,
        data: (r || {}).data
      })
    );
  };
});
c.forEach(["post", "put", "patch"], function(e) {
  function n(r) {
    return function(o, i, a) {
      return this.request(
        I(a || {}, {
          method: e,
          headers: r ? {
            "Content-Type": "multipart/form-data"
          } : {},
          url: o,
          data: i
        })
      );
    };
  }
  v.prototype[e] = n(), v.prototype[e + "Form"] = n(!0);
});
let Rr = class Ct {
  constructor(e) {
    if (typeof e != "function")
      throw new TypeError("executor must be a function.");
    let n;
    this.promise = new Promise(function(o) {
      n = o;
    });
    const r = this;
    this.promise.then((s) => {
      if (!r._listeners) return;
      let o = r._listeners.length;
      for (; o-- > 0; )
        r._listeners[o](s);
      r._listeners = null;
    }), this.promise.then = (s) => {
      let o;
      const i = new Promise((a) => {
        r.subscribe(a), o = a;
      }).then(s);
      return i.cancel = function() {
        r.unsubscribe(o);
      }, i;
    }, e(function(o, i, a) {
      r.reason || (r.reason = new Y(o, i, a), n(r.reason));
    });
  }
  /**
   * Throws a `CanceledError` if cancellation has been requested.
   */
  throwIfRequested() {
    if (this.reason)
      throw this.reason;
  }
  /**
   * Subscribe to the cancel signal
   */
  subscribe(e) {
    if (this.reason) {
      e(this.reason);
      return;
    }
    this._listeners ? this._listeners.push(e) : this._listeners = [e];
  }
  /**
   * Unsubscribe from the cancel signal
   */
  unsubscribe(e) {
    if (!this._listeners)
      return;
    const n = this._listeners.indexOf(e);
    n !== -1 && this._listeners.splice(n, 1);
  }
  toAbortSignal() {
    const e = new AbortController(), n = (r) => {
      e.abort(r);
    };
    return this.subscribe(n), e.signal.unsubscribe = () => this.unsubscribe(n), e.signal;
  }
  /**
   * Returns an object that contains a new `CancelToken` and a function that, when called,
   * cancels the `CancelToken`.
   */
  static source() {
    let e;
    return {
      token: new Ct(function(s) {
        e = s;
      }),
      cancel: e
    };
  }
};
function Er(t) {
  return function(n) {
    return t.apply(null, n);
  };
}
function Sr(t) {
  return c.isObject(t) && t.isAxiosError === !0;
}
const Ae = {
  Continue: 100,
  SwitchingProtocols: 101,
  Processing: 102,
  EarlyHints: 103,
  Ok: 200,
  Created: 201,
  Accepted: 202,
  NonAuthoritativeInformation: 203,
  NoContent: 204,
  ResetContent: 205,
  PartialContent: 206,
  MultiStatus: 207,
  AlreadyReported: 208,
  ImUsed: 226,
  MultipleChoices: 300,
  MovedPermanently: 301,
  Found: 302,
  SeeOther: 303,
  NotModified: 304,
  UseProxy: 305,
  Unused: 306,
  TemporaryRedirect: 307,
  PermanentRedirect: 308,
  BadRequest: 400,
  Unauthorized: 401,
  PaymentRequired: 402,
  Forbidden: 403,
  NotFound: 404,
  MethodNotAllowed: 405,
  NotAcceptable: 406,
  ProxyAuthenticationRequired: 407,
  RequestTimeout: 408,
  Conflict: 409,
  Gone: 410,
  LengthRequired: 411,
  PreconditionFailed: 412,
  PayloadTooLarge: 413,
  UriTooLong: 414,
  UnsupportedMediaType: 415,
  RangeNotSatisfiable: 416,
  ExpectationFailed: 417,
  ImATeapot: 418,
  MisdirectedRequest: 421,
  UnprocessableEntity: 422,
  Locked: 423,
  FailedDependency: 424,
  TooEarly: 425,
  UpgradeRequired: 426,
  PreconditionRequired: 428,
  TooManyRequests: 429,
  RequestHeaderFieldsTooLarge: 431,
  UnavailableForLegalReasons: 451,
  InternalServerError: 500,
  NotImplemented: 501,
  BadGateway: 502,
  ServiceUnavailable: 503,
  GatewayTimeout: 504,
  HttpVersionNotSupported: 505,
  VariantAlsoNegotiates: 506,
  InsufficientStorage: 507,
  LoopDetected: 508,
  NotExtended: 510,
  NetworkAuthenticationRequired: 511,
  WebServerIsDown: 521,
  ConnectionTimedOut: 522,
  OriginIsUnreachable: 523,
  TimeoutOccurred: 524,
  SslHandshakeFailed: 525,
  InvalidSslCertificate: 526
};
Object.entries(Ae).forEach(([t, e]) => {
  Ae[e] = t;
});
function _t(t) {
  const e = new v(t), n = at(v.prototype.request, e);
  return c.extend(n, v.prototype, e, { allOwnKeys: !0 }), c.extend(n, e, null, { allOwnKeys: !0 }), n.create = function(s) {
    return _t(I(t, s));
  }, n;
}
const E = _t(Q);
E.Axios = v;
E.CanceledError = Y;
E.CancelToken = Rr;
E.isCancel = Rt;
E.VERSION = xt;
E.toFormData = fe;
E.AxiosError = g;
E.Cancel = E.CanceledError;
E.all = function(e) {
  return Promise.all(e);
};
E.spread = Er;
E.isAxiosError = Sr;
E.mergeConfig = I;
E.AxiosHeaders = C;
E.formToJSON = (t) => bt(c.isHTMLForm(t) ? new FormData(t) : t);
E.getAdapter = At.getAdapter;
E.HttpStatusCode = Ae;
E.default = E;
const {
  Axios: rs,
  AxiosError: ss,
  CanceledError: os,
  isCancel: is,
  CancelToken: as,
  VERSION: cs,
  all: ls,
  Cancel: us,
  isAxiosError: fs,
  spread: ds,
  toFormData: ps,
  AxiosHeaders: hs,
  HttpStatusCode: ms,
  formToJSON: gs,
  getAdapter: ys,
  mergeConfig: ws
} = E, Or = {
  form: "application/x-www-form-urlencoded",
  json: "application/json",
  data: "multipart/form-data"
}, Tr = ["put", "post", "patch"], rt = "Local-Request-Id", Ar = 100, xr = 300;
class Cr {
  constructor(e = {}) {
    _(this, "axios");
    _(this, "settings");
    _(this, "records", {});
    _(this, "isLoading", !1);
    _(this, "stopSkipWarn");
    _(this, "showLoading");
    _(this, "showError");
    this.settings = Object.assign({ type: "form" }, e.settings || {});
    const n = me(e, [
      "settings",
      "query"
    ]);
    this.axios = E.create(
      W(
        {
          headers: {
            "Content-Type": "application/x-www-form-urlencoded"
          },
          timeout: 120 * 1e3
        },
        n
      )
    ), this.setupSkipWarn(this.settings), this.showLoading = Bt(this.openLoading.bind(this), Ar), this.showError = vt(this._showError.bind(this), xr, {
      leading: !0,
      trailing: !1
    });
  }
  setConfig(e = {}) {
    this.settings = W(this.settings, e.settings || {});
    const n = me(e, [
      "settings",
      "query"
    ]);
    this.axios.defaults = W(this.axios.defaults, n), this.setupSkipWarn(this.settings);
  }
  cancel(e, n = "请求已取消") {
    if (e) {
      const r = this.records[e];
      if (!r) return;
      r.source.cancel(n);
    } else
      for (const r of Object.values(this.records))
        r.source.cancel(n);
  }
  createHeaders(e, n, r) {
    const s = n.injectHeaders ? typeof n.headers == "function" ? n.headers(e, r, n) : n.headers || {} : {}, o = {
      "Content-Type": Or[n.type || "form"],
      ...r.headers,
      ...s
    };
    return n.skipWarn && (o[rt] = e), o;
  }
  isJsonType(e) {
    return Object.entries(e).some(([n, r]) => n.toLowerCase() === "content-type" && String(r).includes("application/json"));
  }
  toFormData(e, n = "data") {
    if (e instanceof FormData || e instanceof URLSearchParams)
      return e;
    const r = n === "data" ? new FormData() : new URLSearchParams();
    return Object.entries(e).forEach(([s, o]) => {
      r.append(s, o);
    }), r;
  }
  createSendData(e, n, r, s, o = {}) {
    const { type: i, skipWarn: a } = e, { name: f = "skipWarn" } = a || {};
    let { data: u, params: l = {}, method: p = "get" } = n;
    const y = s ? { [f]: !0 } : {};
    return Tr.includes(p.toLowerCase()) ? (u = Object.assign(u || {}, y), u = i !== "json" || !this.isJsonType(r) ? this.toFormData(u, i) : u, l = {
      ...o
    }) : i === "form" ? l = {
      ...u || {},
      ...o,
      ...y
    } : (u && (i !== "json" || !this.isJsonType(r)) && (u = this.toFormData(u, i)), l = {
      ...o,
      ...y
    }), {
      data: u,
      params: l
    };
  }
  createUrl(e) {
    let { url: n, params: r } = e;
    if (n) {
      let s = It(n) ? new URL(n).origin : "";
      const o = s ? n.replace(s, "") : n;
      try {
        const i = qt(o, {
          encode: encodeURIComponent
        });
        return s + i(r || {});
      } catch {
        console.warn("createUrl", "pathToRegexpCompile error", n);
      }
    }
    return n;
  }
  openLoading(e) {
    const { loading: n, showLoading: r } = e;
    n && r && Object.keys(this.records).length > 0 && (this.isLoading = !0, r());
  }
  closeLoading(e) {
    const { loading: n, hideLoading: r } = e;
    if (!n) return;
    this.isLoading = !1;
    const s = Object.keys(this.records);
    r && s.length === 0 && (this.isLoading = !1, r());
  }
  _showError(e, n) {
    const { failMessage: r, showError: s } = e;
    if (r && s) {
      const o = n?.response?.data, i = o?.message || o?.msg || n?.message || n?.msg || "未知错误";
      s(i, n);
    }
  }
  validResponse(e, n) {
    const { validSuccess: r, validate: s } = e;
    return r && s ? !!s(n) : !0;
  }
  isSkipWarnResponse(e) {
    return !!e.promise;
  }
  send(e = {}, n = !1) {
    const r = W({}, this.settings, e.settings || {}), s = e.query || {}, o = me(e, [
      "settings",
      "query"
    ]), i = Mt(!1), a = E.CancelToken.source();
    this.records[i] = { settings: r, config: o, source: a };
    const f = this.createUrl(o), u = this.createHeaders(i, r, o), { data: l, params: p } = this.createSendData(
      r,
      o,
      u,
      n,
      s
    );
    return this.showLoading(r), new Promise((y, R) => {
      this.axios({
        cancelToken: a.token,
        ...o,
        url: f,
        headers: u,
        data: l,
        params: p
      }).then((d) => this.isSkipWarnResponse(d) ? y(d.promise) : this.validResponse(r, d) ? y(r.originResponse ? d : d.data?.data) : (this.showError(r, d.data), R(d.data))).catch((d) => (this.showError(r, d), R(d))).finally(() => {
        delete this.records[i], this.closeLoading(r);
      });
    });
  }
  useResponse(e, n) {
    const { response: r } = this.axios.interceptors, s = r.use(e, n);
    return () => r.eject(s);
  }
  useRequest(e, n) {
    const { request: r } = this.axios.interceptors, s = r.use(e, n);
    return () => r.eject(s);
  }
  setupSkipWarn(e) {
    if (this.stopSkipWarn && (this.stopSkipWarn(), this.stopSkipWarn = void 0), !e.skipWarn) return;
    const { code: n, executor: r, callback: s, complete: o } = e.skipWarn;
    this.stopSkipWarn = this.useResponse((i) => {
      const f = (i.config.headers || {})[rt], u = this.records[f];
      if (!u) return i;
      const { data: l } = i;
      if (!l || typeof l != "object") return i;
      if (l?.code === n) {
        s && s(i);
        const p = new Promise(r).then(() => this.send(
          {
            ...u.config,
            settings: u.settings
          },
          !0
        ));
        p.catch((y) => y).finally(() => {
          o && o();
        }), i.promise = p;
      }
      return i;
    });
  }
}
function _r(t = {}) {
  const e = new Cr(t), n = e.send.bind(e), r = e.cancel.bind(e), s = e.setConfig.bind(e), o = e.useRequest.bind(e), i = e.useResponse.bind(e);
  return Object.assign(n, {
    ...e,
    instance: e,
    send: n,
    cancel: r,
    setConfig: s,
    useRequest: o,
    useResponse: i
  });
}
const Pt = _r({
  settings: {
    injectHeaders: !0,
    loading: !0,
    originResponse: !0
  }
});
function Pr(t, e = Pt) {
  const n = typeof t == "string" ? { url: t } : t;
  return (r, s) => e.send(W({}, n, s || {}, { data: r }));
}
function bs(t, e = Pt) {
  const n = {};
  for (const [r, s] of Object.entries(t))
    n[r] = Pr(s, e);
  return n;
}
function Rs(t, e) {
  const n = ge(null), r = ge(), s = ge(!0), o = () => {
    s.value = !0, r.value = void 0, (typeof t == "function" ? t() : t).then((a) => {
      n.value = e ? e(a) : a;
    }).catch((a) => {
      r.value = a;
    }).finally(() => {
      s.value = !1;
    });
  };
  return o(), {
    data: n,
    error: r,
    loading: s,
    reload: o
  };
}
const V = typeof window < "u", Es = (t) => new Promise((e, n) => {
  const r = new FileReader();
  r.readAsDataURL(t), r.onload = () => {
    e(r.result);
  }, r.onerror = (s) => {
    n(s);
  };
});
function Ss(t) {
  const e = {};
  return t ? (t.forEach((n, r) => {
    e[r] = typeof n == "string" ? decodeURIComponent(n) : n;
  }), e) : {};
}
function Os(t) {
  const e = t.split(","), n = e[0].match(/:(.*?);/)?.[1], r = atob(e[1]);
  let s = r.length;
  const o = new Uint8Array(s);
  for (; s--; )
    o[s] = r.charCodeAt(s);
  return new Blob([o], { type: n });
}
function Ts(t, e) {
  const n = t;
  return n.lastModified = Date.now(), n.lastModifiedDate = /* @__PURE__ */ new Date(), n.name = e, n;
}
const As = (t) => V ? window.requestAnimationFrame(t) : setTimeout(t, 16), xs = (t) => V ? window.cancelAnimationFrame(t) : clearTimeout(t);
class Lr {
  constructor(e = {}) {
    _(this, "options", {
      type: "cache",
      expired: 0,
      prefix: "__VTJ_"
    });
    _(this, "caches", {});
    _(this, "types");
    this.types = {
      local: V ? window.localStorage : this.caches,
      session: V ? window.sessionStorage : this.caches,
      cache: this.caches
    }, this.config(e);
  }
  config(e = {}) {
    this.options = Object.assign(this.options, e);
  }
  save(e, n, r = {}) {
    const { type: s, expired: o, prefix: i } = { ...this.options, ...r }, a = Date.now(), f = i + e, u = this.types[s] || this.caches, l = {
      value: n,
      timestamp: a,
      expired: o
    };
    u === this.caches ? u[f] = l : u.setItem(f, JSON.stringify(l));
  }
  get(e, n = {}) {
    const { type: r, prefix: s } = { ...this.options, ...n }, o = s + e, i = this.types[r] || this.caches;
    let a;
    if (i === this.caches)
      a = i[o];
    else {
      const y = i.getItem(o);
      if (y)
        try {
          a = JSON.parse(y);
        } catch (R) {
          console.warn("[Storage] JSON.parse", R);
        }
    }
    if (!a) return null;
    const { value: f, timestamp: u, expired: l } = a;
    return l > 0 && u + l < Date.now() ? (this.remove(e, n), null) : f;
  }
  remove(e, n = {}) {
    const { type: r, prefix: s } = { ...this.options, ...n }, o = this.types[r] || this.caches, i = s + e;
    o === this.caches ? delete o[i] : o.removeItem(i);
  }
  clear(e = {}) {
    const { type: n } = { ...this.options, ...e }, r = this.types[n] || this.caches;
    r === this.caches ? this.caches = {} : r.clear();
  }
}
const Cs = new Lr();
function Lt(t) {
  return t && t.__esModule && Object.prototype.hasOwnProperty.call(t, "default") ? t.default : t;
}
var z = { exports: {} }, Nr = z.exports, st;
function Ur() {
  return st || (st = 1, (function(t, e) {
    (function(n, r) {
      r(e, t);
    })(Nr, function(n, r) {
      var s = {
        timeout: 5e3,
        jsonpCallback: "callback"
      };
      function o() {
        return "jsonp_" + Date.now() + "_" + Math.ceil(Math.random() * 1e5);
      }
      function i(u) {
        try {
          delete window[u];
        } catch {
          window[u] = void 0;
        }
      }
      function a(u) {
        var l = document.getElementById(u);
        l && document.getElementsByTagName("head")[0].removeChild(l);
      }
      function f(u) {
        var l = arguments.length <= 1 || arguments[1] === void 0 ? {} : arguments[1], p = u, y = l.timeout || s.timeout, R = l.jsonpCallback || s.jsonpCallback, d = void 0;
        return new Promise(function(m, h) {
          var w = l.jsonpCallbackFunction || o(), S = R + "_" + w;
          window[w] = function(O) {
            m({
              ok: !0,
              // keep consistent with fetch API
              json: function() {
                return Promise.resolve(O);
              }
            }), d && clearTimeout(d), a(S), i(w);
          }, p += p.indexOf("?") === -1 ? "?" : "&";
          var b = document.createElement("script");
          b.setAttribute("src", "" + p + R + "=" + w), l.charset && b.setAttribute("charset", l.charset), l.nonce && b.setAttribute("nonce", l.nonce), l.referrerPolicy && b.setAttribute("referrerPolicy", l.referrerPolicy), l.crossorigin && b.setAttribute("crossorigin", "true"), b.id = S, document.getElementsByTagName("head")[0].appendChild(b), d = setTimeout(function() {
            h(new Error("JSONP request to " + u + " timed out")), i(w), a(S), window[w] = function() {
              i(w);
            };
          }, y), b.onerror = function() {
            h(new Error("JSONP request to " + u + " failed")), i(w), a(S), d && clearTimeout(d);
          };
        });
      }
      r.exports = f;
    });
  })(z, z.exports)), z.exports;
}
var Fr = Ur();
const jr = /* @__PURE__ */ Lt(Fr);
function kr(t) {
  if (V) {
    const { protocol: e, host: n, pathname: r } = location;
    return `${e}//${n}${t ? r : ""}`;
  } else
    return null;
}
function Dr(t = "") {
  const e = t.match(Ht);
  return e ? e[0] : "";
}
function Nt(t) {
  const e = [];
  for (const n in t)
    Object.prototype.hasOwnProperty.call(t, n) && e.push([n, encodeURIComponent(t[n])].join("="));
  return e.join("&");
}
function xe(t, e, n) {
  const r = {};
  t = (t || location.search).replace(/^[^]*\?/, ""), e = e || "&", n = n || "=";
  let s;
  const o = new RegExp(
    "(?:^|\\" + e + ")([^\\" + n + "\\" + e + "]+)(?:\\" + n + "([^\\" + e + "]*))?",
    "g"
  );
  for (; (s = o.exec(t)) !== null; )
    s[1] !== t && (r[decodeURIComponent(s[1])] = decodeURIComponent(s[2] || ""));
  return r;
}
function Ut(t, e) {
  e = typeof e == "string" ? xe(e) : e;
  const n = t.split("?")[0], r = xe(t), s = Object.assign({}, r, e), o = Nt(s);
  return o ? [n, o].join("?") : t;
}
const _s = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  append: Ut,
  getCurrentHost: kr,
  getHost: Dr,
  parse: xe,
  stringify: Nt
}, Symbol.toStringTag, { value: "Module" }));
async function Ps(t, e = {}) {
  const { query: n = {} } = e;
  t.includes("${") && (t = $t(t)(n || {}));
  const r = Ut(t, n);
  return await (await jr(r, e)).json();
}
var Ee, ot;
function Br() {
  if (ot) return Ee;
  ot = 1, Ee = function(s, o, i) {
    var a = document.head || document.getElementsByTagName("head")[0], f = document.createElement("script");
    typeof o == "function" && (i = o, o = {}), o = o || {}, i = i || function() {
    }, f.type = o.type || "text/javascript", f.charset = o.charset || "utf8", f.async = "async" in o ? !!o.async : !0, f.src = s, o.attrs && t(f, o.attrs), o.text && (f.text = "" + o.text);
    var u = "onload" in f ? e : n;
    u(f, i), f.onload || e(f, i), a.appendChild(f);
  };
  function t(r, s) {
    for (var o in s)
      r.setAttribute(o, s[o]);
  }
  function e(r, s) {
    r.onload = function() {
      this.onerror = this.onload = null, s(null, r);
    }, r.onerror = function() {
      this.onerror = this.onload = null, s(new Error("Failed to load " + this.src), r);
    };
  }
  function n(r, s) {
    r.onreadystatechange = function() {
      this.readyState != "complete" && this.readyState != "loaded" || (this.onreadystatechange = null, s(null, r));
    };
  }
  return Ee;
}
var vr = Br();
const Ir = /* @__PURE__ */ Lt(vr);
function Ls(t, e = {}) {
  return new Promise((n, r) => {
    const { library: s } = e;
    Ir(t, e, (o, i) => {
      o ? r(o) : n(s ? window[s] : void 0);
    });
  });
}
const it = { debug: -1, log: 0, info: 0, warn: 1, error: 2 }, qr = function(t, e, n, r) {
  return function(...s) {
    if (e && it[e] <= it[t] && // @ts-ignore
    console[t].apply && (r === "*" || n.startsWith(r)))
      return console[t].apply(console, Mr(s, n));
  };
};
function Mr(t, e) {
  return e !== "*" && (typeof t[0] == "string" ? t[0] = `[${e}] ${t[0]}` : t = ["[" + e + "]"].concat(t)), t;
}
function Hr(t, e) {
  if (!t)
    return {
      targetLevel: e.level,
      targetBizName: e.bizName
    };
  if (~t.indexOf(":")) {
    const n = t.split(":");
    return {
      targetLevel: n[0],
      targetBizName: n[1]
    };
  }
  return {
    targetLevel: t,
    targetBizName: "*"
  };
}
const $r = {
  level: "warn",
  bizName: "*"
};
class Jr {
  constructor(e) {
    _(this, "config");
    _(this, "options");
    this.options = { ...$r, ...e };
    const n = typeof location < "u" ? location : {}, r = (/__(?:logConf|logLevel)__=([^#/&]*)/.exec(
      n.href
    ) || [])[1];
    this.config = Hr(r, e);
  }
  _log(e) {
    const { targetLevel: n, targetBizName: r } = this.config, { bizName: s } = this.options;
    return qr(e, n, s, r);
  }
  debug(...e) {
    return this._log("debug")(...e);
  }
  log(...e) {
    return this._log("log")(...e);
  }
  info(...e) {
    return this._log("info")(...e);
  }
  warn(...e) {
    return this._log("warn")(...e);
  }
  error(...e) {
    return this._log("error")(...e);
  }
}
function Wr(t) {
  return new Jr(t);
}
const Ns = Wr({ level: "log", bizName: "VTJ" });
/*! js-cookie v3.0.5 | MIT */
function re(t) {
  for (var e = 1; e < arguments.length; e++) {
    var n = arguments[e];
    for (var r in n)
      t[r] = n[r];
  }
  return t;
}
var zr = {
  read: function(t) {
    return t[0] === '"' && (t = t.slice(1, -1)), t.replace(/(%[\dA-F]{2})+/gi, decodeURIComponent);
  },
  write: function(t) {
    return encodeURIComponent(t).replace(
      /%(2[346BF]|3[AC-F]|40|5[BDE]|60|7[BCD])/g,
      decodeURIComponent
    );
  }
};
function Ce(t, e) {
  function n(s, o, i) {
    if (!(typeof document > "u")) {
      i = re({}, e, i), typeof i.expires == "number" && (i.expires = new Date(Date.now() + i.expires * 864e5)), i.expires && (i.expires = i.expires.toUTCString()), s = encodeURIComponent(s).replace(/%(2[346B]|5E|60|7C)/g, decodeURIComponent).replace(/[()]/g, escape);
      var a = "";
      for (var f in i)
        i[f] && (a += "; " + f, i[f] !== !0 && (a += "=" + i[f].split(";")[0]));
      return document.cookie = s + "=" + t.write(o, s) + a;
    }
  }
  function r(s) {
    if (!(typeof document > "u" || arguments.length && !s)) {
      for (var o = document.cookie ? document.cookie.split("; ") : [], i = {}, a = 0; a < o.length; a++) {
        var f = o[a].split("="), u = f.slice(1).join("=");
        try {
          var l = decodeURIComponent(f[0]);
          if (i[l] = t.read(u, l), s === l)
            break;
        } catch {
        }
      }
      return s ? i[s] : i;
    }
  }
  return Object.create(
    {
      set: n,
      get: r,
      remove: function(s, o) {
        n(
          s,
          "",
          re({}, o, {
            expires: -1
          })
        );
      },
      withAttributes: function(s) {
        return Ce(this.converter, re({}, this.attributes, s));
      },
      withConverter: function(s) {
        return Ce(re({}, this.converter, s), this.attributes);
      }
    },
    {
      attributes: { value: Object.freeze(e) },
      converter: { value: Object.freeze(t) }
    }
  );
}
var Fe = Ce(zr, { path: "/" });
function Vr(t, e, n) {
  Fe.set(t, e, n);
}
function Kr(t) {
  return Fe.get(t);
}
function Xr(t, e) {
  Fe.remove(t, e);
}
const Us = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  get: Kr,
  remove: Xr,
  set: Vr
}, Symbol.toStringTag, { value: "Module" }));
function Fs(t, e = "") {
  const n = document.createElement("a");
  n.download = e, n.href = t, n.target = "_blank", n.click();
}
function Ft(t, e = "", n) {
  const r = new Blob([t], { type: n }), s = document.createElement("a");
  s.download = e, s.style.display = "none", s.href = URL.createObjectURL(r), s.click(), URL.revokeObjectURL(s.href);
}
async function js(t, e = "", n) {
  return fetch(t, { credentials: "include" }).then(async (r) => {
    const s = await r.blob();
    return Ft(s, e, n), s;
  });
}
function ks(t, e = "") {
  const n = JSON.stringify(t);
  Ft(n, e, "application/json");
}
function Ds() {
  const t = navigator.userAgent;
  let e = "Unknown", n = "Unknown", r = "Unknown", s = "Unknown", o = !1;
  if (/Windows NT/i.test(t)) {
    e = "Windows";
    const a = {
      "12.0": "12",
      "11.0": "11",
      "10.0": "10",
      "6.3": "8.1",
      "6.2": "8",
      "6.1": "7",
      "6.0": "Vista",
      "5.2": "XP 64-bit",
      "5.1": "XP"
    }, f = t.match(/Windows NT (\d+\.\d+)/);
    f && (n = a[f[1]] || f[1]);
  } else if (/Mac OS X/i.test(t)) {
    e = "Mac OS";
    const a = t.match(/Mac OS X (\d+[._]\d+[._]?\d*)/);
    a && (n = a[1].replace(/_/g, "."));
  } else if (/(iPhone|iPad|iPod)/i.test(t)) {
    e = "iOS";
    const a = t.match(/OS (\d+[_\.]\d+[_\.]?\d*)/);
    a && (n = a[1].replace(/_/g, "."));
  } else if (/Android/i.test(t)) {
    e = "Android";
    const a = t.match(/Android (\d+\.\d+)/);
    a && (n = a[1]);
  } else /Linux/i.test(t) && (e = "Linux");
  const i = t.match(/(Edge|Edg|Edga|EdgA)\/(\d+)/i);
  if (i)
    r = "Microsoft Edge", s = i[2];
  else {
    const a = t.match(/Firefox\/(\d+)/i);
    if (a)
      r = "Firefox", s = a[1];
    else {
      const f = t.match(/(Opera|OPR)\/(\d+)/i);
      if (f)
        r = "Opera", s = f[2];
      else {
        const u = t.match(/Chrome\/(\d+)/i);
        if (u)
          r = "Chrome", s = u[1];
        else {
          const l = t.match(/Version\/(\d+\.\d+)/i);
          if (l && /Safari/i.test(t))
            r = "Safari", s = l[1];
          else {
            const p = t.match(/(MSIE |Trident.*rv:)(\d+)/i);
            p && (r = "Internet Explorer", s = p[2]);
          }
        }
      }
    }
  }
  return o = // 移动设备通用检测
  /(iPhone|iPod|iPad|Android|Windows Phone|Mobile)/i.test(t) || // 根据已识别的操作系统补充判断
  ["iOS", "Android"].includes(e), /(iPad|Tablet|Android(?!.*Mobile))/i.test(t) && (o = !0), {
    os: e,
    osVersion: n,
    browser: r,
    browserVersion: s,
    isMobile: o
  };
}
export {
  rt as LOCAL_REQUEST_ID,
  Jr as Logger,
  Cr as Request,
  Lr as Storage,
  Zr as VTJ_UTILS_VERSION,
  E as axios,
  Ts as blobToFile,
  xs as cAF,
  Us as cookie,
  Pr as createApi,
  bs as createApis,
  _r as createRequest,
  Os as dataURLtoBlob,
  Ft as downloadBlob,
  ks as downloadJson,
  js as downloadRemoteFile,
  Fs as downloadUrl,
  Es as fileToBase64,
  Ss as formDataToJson,
  Ds as getClientInfo,
  Wr as getLogger,
  V as isClient,
  Ps as jsonp,
  Ls as loadScript,
  Ns as logger,
  As as rAF,
  Pt as request,
  Cs as storage,
  _s as url,
  Rs as useApi
};
